---
title: "SummarySporeTraitAnalysis"
author: "Carlos"
date: "22/04/2020"
output:
  html_document: default
  word_document: default
---


```{r, message=FALSE, warning=FALSE}
library(smatr)
library(performance)
library(tidyverse)
library(lme4)
source("MatchingSpore_FunctionData.R")
```

# Statistical analysis


Correct for phylogeny using as a random factor (spore type as main factors)
```{r}

model0<-
lmer(log10(SporeVolume)~1+(1|phylum/class/order/family),data = 
       To_Analysis#[which(To_Analysis$SporeType=="Meiospores"),]
     )

model1<-
lmer(log10(SporeVolume)~SporeType+(1|phylum/class/order/family),data = 
       To_Analysis)


summary(model0)

```


```{r}
summary(model1)
```

"The marginal r-squared considers only the variance of the fixed effects, while the conditional r-squared takes both the fixed and random effects into account."
```{r}
r2_nakagawa(model1)
```


## Importanc of lifestyle in explaining interpecific spore varaiton (after correcting for phylogeny using order as a random factor) for each spore type

Ascospores
```{r}
ascospores_lifestyle<-
  lmer(log10(SporeVolume)~simpleFunct+(1|order),data = 
       Ascospores)

summary(ascospores_lifestyle)
```

```{r}
r2_nakagawa(ascospores_lifestyle)

```

```{r}
tapply(log10(Ascospores$SporeVolume),Ascospores$simpleFunct,mean)
```

Basidiospores
```{r}
#Basidiospores
#table(Basidiospores$simpleFunct)
basidiospore_lifestyle<-
  lmer(log10(SporeVolume)~simpleFunct+(1|order),data = 
       Basidiospores)

summary(basidiospore_lifestyle)

r2_nakagawa(basidiospore_lifestyle)
```


Conidia (Mitospores)
```{r}
conidia_lifestyle<-
  lmer(log10(SporeVolume)~simpleFunct+(1|order),data = 
       Conidia)

summary(conidia_lifestyle)

r2_nakagawa(conidia_lifestyle)
```


## Determinig the relative importance of climate vs life style in explaining interpecific spore variation (From Jeff, sent on September 2020)

```{r, eval=FALSE, echo=FALSE}

# load libraries
library(lme4)
library(car)

## read data (only includes species with a single spore type)
#dat <- read_csv('derivedData/df_species_byPrimerSet.csv')
dat <- read_csv('output/df_species_noPrimers.csv')

## spore volume, establish a common dataset for all models
tmp <- dat %>% 
  select(log10.sporeVolume, Primers.name, Species, phylum:order, host.assoc, ends_with('mean')) %>% 
  filter(complete.cases(.)) %>% 
  mutate_at(vars(ends_with('mean')), scale)

# global model
m1 <- lmer(log10.sporeVolume ~ host.assoc + 
             mat_mean + map_mean + maxsrad_mean + minvapr_mean + 
             (1|order), data=tmp, REML=TRUE)
summary(m1)

# host-only model
m2 <- lmer(log10.sporeVolume ~ host.assoc + (1|order) + (1|Primers.name), data=tmp, REML=TRUE)

# environment-only model
m3 <- lmer(log10.sporeVolume ~ mat_mean + map_mean + maxsrad_mean + minvapr_mean + 
             (1|order) + (1|Primers.name), data=tmp, REML=TRUE)

# compare
anova(m1, m2)
anova(m1, m3)

r2_nakagawa(m1)

# global model
m1 <- lm(log10.sporeVolume ~ order + host.assoc + 
             mat_mean + map_mean + maxsrad_mean + minvapr_mean, data=tmp)
drop1(m1)
summary(m1)

# 

# host-only model
m2 <- lm(log10.sporeVolume ~ order + host.assoc + 
             mat_mean + map_mean + maxsrad_mean + minvapr_mean, data=tmp)

# environment-only model
m3 <- lm(log10.sporeVolume ~ order + host.assoc + 
             mat_mean + map_mean + maxsrad_mean + minvapr_mean, data=tmp)

```



#Combining Jeff and Carlos approach
```{r}
# conidia_lifestyle<-
#   lmer(log10(SporeVolume)~simpleFunct+(1|order),data = 
#        Conidia)
# dat <- read_csv('output/df_species_byPrimerSet.csv')
dat <- read_csv('output/df_species_noPrimers.csv')

dat$names_to_use<-gsub("_"," ",dat$Species)

# trial<-left_join(dat %>% 
#                   select(names_to_use,Primers.name,host.assoc, ends_with('mean')), 
#                  To_Analysis[,c("names_to_use","SporeType","simpleFunct","SporeVolume", "Q_ratio",
#                                 "phylum","class","order")])
trial<-left_join(dat %>% 
                  select(names_to_use, host.assoc, log10.sporeVolume, log10.sporeAspectRatio, ends_with('mean'), n_samples_env), 
                 To_Analysis[,c("names_to_use","SporeType","simpleFunct",
                                "phylum","class","order")])

# this is weird that there are species without assignments
trial<-trial[complete.cases(trial),] %>% 
  filter(!order == 'Not assigned')

table(trial$simpleFunct)
# filter(trial, simpleFunct == 'Human')

# libraries for tables
library(broom)
library(pixiedust)

```


## Spore volume
```{r}

### spore volume

## global model
m1 <- lm(log10.sporeVolume ~ order + simpleFunct + SporeType + 
             alt_mean + mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# # plot to check for nonlinearities
# tmp <- m1$model %>% 
#   mutate_if(is.character, as.factor)
# for(i in 2:ncol(tmp)) plot(tmp$log10.sporeVolume ~ tmp[, i], main=names(tmp)[i]) 

## results for individual variables
m1.var <- vector('list', length=ncol(m1$model))
names(m1.var) <- c('<none>', names(m1$model)[2:ncol(m1$model)])
m1.var[[1]] <- m1$model
for(i in 2:length(m1.var)) m1.var[[i]] <- m1$model[, -i]
m1.aic <- lapply(m1.var, function(x)AIC(m1, lm(log10.sporeVolume ~ ., data=x)))
m1.drp <- data.frame(term = names(m1.var), 
                     Df = sapply(m1.aic, function(x)x[2, 'df']), 
                     AIC = sapply(m1.aic, function(x)x[2, 'AIC']), 
                     AdjR2 = sapply(m1.var, function(x)summary(lm(log10.sporeVolume ~ ., data=x))[['adj.r.squared']])) %>% 
  mutate(type = c('global', rep('subset', n() - 1)), 
         dAIC = AIC - AIC[1], 
         dAdjR2 = AdjR2 - AdjR2[1]) %>% 
  group_by(type) %>% 
  arrange(desc(dAIC), .by_group=TRUE) %>% 
  ungroup() %>% 
  select(term, Df, AIC, dAIC, AdjR2, dAdjR2)

## results for variable groups
# drop taxonomy
m2.tax <- lm(log10.sporeVolume ~ simpleFunct + SporeType + 
               alt_mean + mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop function
m2.fun <- lm(log10.sporeVolume ~ order + SporeType + 
               alt_mean + mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop spore type
m2.typ <- lm(log10.sporeVolume ~ order + simpleFunct + 
               alt_mean + mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop environment
m2.env <- lm(log10.sporeVolume ~ order + simpleFunct + SporeType, data=trial)
# make table
m2.drp <- data.frame(term=c('<none>', 'taxonomy', 'simpleFunct', 'SporeType', 'climate'), 
                     df=AIC(m1, m2.tax, m2.fun, m2.typ, m2.env)[['df']], 
                     AIC=AIC(m1, m2.tax, m2.fun, m2.typ, m2.env)[['AIC']], 
                     AdjR2=sapply(list(m1, m2.tax, m2.fun, m2.typ, m2.env), function(x)summary(x)[['adj.r.squared']]), 
                     row.names=NULL) %>% 
  mutate(type = c('<none>', rep('subset', n() - 1)), 
         dAIC = AIC - AIC[1], 
         dAdjR2 = AdjR2 - AdjR2[1]) %>% 
  group_by(type) %>% 
  arrange(desc(dAIC), .by_group=TRUE) %>% 
  ungroup() %>% 
  select(term, df, AIC, dAIC, AdjR2, dAdjR2)

# Table 1
dust(m1.drp, caption='Table X. Importance of individual variables in linear models for explaining variation among species in spore volume. Degrees of freedom (df), Akaike\'s Information Criterion (AIC), delta AIC (dAIC), adjusted R^2 (Adj. R2) and delta adjusted R^2 (dAdj. R2) were calculated by dropping terms from a global model containing all terms (indicated in italics).') %>%
  sprinkle_print_method('markdown')  %>%
  sprinkle(rows=1, italic=TRUE) %>%
  sprinkle(cols=c(3, 4), round=1) %>%
  sprinkle(cols=c(5, 6), round=3)

# Table 2
dust(m2.drp, caption='Table X. Importance of variable groupss in linear models for explaining variation among species in spore volume. Degrees of freedom (df), Akaike\'s Information Criterion (AIC), delta AIC (dAIC), adjusted R^2 (Adj. R2) and delta adjusted R^2 (dAdj. R2) were calculated by dropping groups of terms from a global model containing all terms (indicated in italics). ') %>%
  sprinkle_print_method('markdown')  %>%
  sprinkle(rows=1, italic=TRUE) %>%
  sprinkle(cols=c(3, 4), round=1) %>%
  sprinkle(cols=c(5, 6), round=3)
  
```

## Spore aspect ratio
```{r}

### spore aspect ratio

## global model
m1 <- lm(log10.sporeAspectRatio ~ order + simpleFunct + SporeType + 
             alt_mean + mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# # plot to check for nonlinearities
# tmp <- m1$model %>%
#   mutate_if(is.character, as.factor)
# for(i in 2:ncol(tmp)) plot(tmp$log10.sporeAspectRatio ~ tmp[, i], main=names(tmp)[i])

## results for individual variables
m1.var <- vector('list', length=ncol(m1$model))
names(m1.var) <- c('<none>', names(m1$model)[2:ncol(m1$model)])
m1.var[[1]] <- m1$model
for(i in 2:length(m1.var)) m1.var[[i]] <- m1$model[, -i]
m1.aic <- lapply(m1.var, function(x)AIC(m1, lm(log10.sporeAspectRatio ~ ., data=x)))
m1.drp <- data.frame(term = names(m1.var), 
                     Df = sapply(m1.aic, function(x)x[2, 'df']), 
                     AIC = sapply(m1.aic, function(x)x[2, 'AIC']), 
                     AdjR2 = sapply(m1.var, function(x)summary(lm(log10.sporeAspectRatio ~ ., data=x))[['adj.r.squared']])) %>% 
  mutate(type = c('global', rep('subset', n() - 1)), 
         dAIC = AIC - AIC[1], 
         dAdjR2 = AdjR2 - AdjR2[1]) %>% 
  group_by(type) %>% 
  arrange(desc(dAIC), .by_group=TRUE) %>% 
  ungroup() %>% 
  select(term, Df, AIC, dAIC, AdjR2, dAdjR2)

## results for variable groups
# drop taxonomy
m2.tax <- lm(log10.sporeAspectRatio ~ simpleFunct + SporeType + 
               alt_mean + mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop function
m2.fun <- lm(log10.sporeAspectRatio ~ order + SporeType + 
               alt_mean + mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop spore type
m2.typ <- lm(log10.sporeAspectRatio ~ order + simpleFunct + 
               alt_mean + mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop environment
m2.env <- lm(log10.sporeAspectRatio ~ order + simpleFunct + SporeType, data=trial)
# make table
m2.drp <- data.frame(term=c('<none>', 'taxonomy', 'simpleFunct', 'SporeType', 'climate'), 
                     df=AIC(m1, m2.tax, m2.fun, m2.typ, m2.env)[['df']], 
                     AIC=AIC(m1, m2.tax, m2.fun, m2.typ, m2.env)[['AIC']], 
                     AdjR2=sapply(list(m1, m2.tax, m2.fun, m2.typ, m2.env), function(x)summary(x)[['adj.r.squared']]), 
                     row.names=NULL) %>% 
  mutate(type = c('global', rep('subset', n() - 1)), 
         dAIC = AIC - AIC[1], 
         dAdjR2 = AdjR2 - AdjR2[1]) %>% 
  group_by(type) %>% 
  arrange(desc(dAIC), .by_group=TRUE) %>% 
  ungroup() %>% 
  select(term, df, AIC, dAIC, AdjR2, dAdjR2)

# Table 1
dust(m1.drp, caption='Table X. Importance of individual variables in linear models for explaining variation among species in spore shape. Degrees of freedom (df), Akaike\'s Information Criterion (AIC), delta AIC (dAIC), adjusted R^2 (Adj. R2) and delta adjusted R^2 (dAdj. R2) were calculated by dropping terms from a global model containing all terms (indicated in italics).') %>%
  sprinkle_print_method('markdown')  %>%
  sprinkle(rows=1, italic=TRUE) %>%
  sprinkle(cols=c(3, 4), round=1) %>%
  sprinkle(cols=c(5, 6), round=3)

# Table 2
dust(m2.drp, caption='Table X. Importance of variable groupss in linear models for explaining variation among species in spore shape. Degrees of freedom (df), Akaike\'s Information Criterion (AIC), delta AIC (dAIC), adjusted R^2 (Adj. R2) and delta adjusted R^2 (dAdj. R2) were calculated by dropping groups of terms from a global model containing all terms (indicated in italics). ') %>%
  sprinkle_print_method('markdown')  %>%
  sprinkle(rows=1, italic=TRUE) %>%
  sprinkle(cols=c(3, 4), round=1) %>%
  sprinkle(cols=c(5, 6), round=3)

```


# Figures

## Figure 1 (further modified in inkspace)

### Figure 1 Phylogenetic Tree

```{r}
library(ape)
library(ggtree)

#devtools::install_github("ropenscilabs/datastorr")
#devtools::install_github("wcornwell/taxonlookup")

library(taxonlookup)

# phylogeny from https://www.nature.com/articles/s41467-018-07849-9
tt<-read.nexus("phylo/Lutzoni_fungi_timetree.nex")
vcapply<-taxonlookup:::vcapply

split<-function(str){
  str_split <- strsplit(str, "[_ ]+")
  vcapply(str_split, "[[", 2L)
}

#generate genus in order lookup table
#fungal_genera<-split(nnn)[1:197]
#write_csv(tibble(genus=fungal_genera),"phylo/genus_names.csv")

#read in genus in order lookup
orders<-read_csv("phylo/orders_phylo.csv")
orders %>% group_by(order) %>%
  summarize(genus=genus[1]) -> only_one_genera_per_order

#getting a tree with one tip per order
nnn<-tt$tip.label
tt$tip.label<-split(nnn)
dropped.tree<-drop.tip(tt,tt$tip.label[which(!tt$tip.label %in% only_one_genera_per_order$genus)])
dup.tips<-names(which(table(dropped.tree$tip.label)>1))
tips<-dropped.tree$tip.label 
dropped.tree2<-drop.tip(dropped.tree,which(duplicated(tips)))
dropped.tree2$tip.label<-orders$order[match(dropped.tree2$tip.label,orders$genus)]



#read in spore data and calculate size and shape variables
#sdf<-read_csv("output/Spore_Database_Fungi.csv")
matched_data<-filter(AllFungi,order%in%dropped.tree2$tip.label)
matched_data$log.length<-log10(matched_data$spore_length)
#matched_data$log_spore_area<-log10(matched_data$spore_length*matched_data$spore_width * pi /4)
matched_data$log_spore_volume<-log10((matched_data$spore_width^2)*matched_data$spore_length*(pi/6))
matched_data$length_divided_by_width<-log10(matched_data$spore_length/matched_data$spore_width)


#check tree
dropped.tree2$tip.label %in% matched_data$order

#relabel to ggtree convention
matched_data$id <- matched_data$order

#drop not assigned taxa; there are multiple spots on tree with non assigned taxa
matched_data<-filter(matched_data,order!="Not assigned")
dropped.tree2<-drop.tip(dropped.tree2,"Not assigned")

#drop small spore size categories for plotting
match2<-filter(matched_data,!SporeName %in% c("bulbil","papulospore"))

#select only necessary data
#temp<-select(match2,id,log_spore_area,SporeName,length_divided_by_width)
temp<-select(match2,
             id, phylum,log_spore_volume, SporeName, length_divided_by_width)

temp$SporeType<-NA
temp$SporeType[which(temp$SporeName=="Ascospores"|temp$SporeName=="Basidiospores")]<-"Meiospores"
temp$SporeType[which(temp$SporeName=="Conidia"|temp$SporeName=="Sporangiospores"|temp$SporeName=="Chlamydospores")]<-"Mitospores"
temp$SporeType[which(temp$SporeName=="Azygospores")]<-"Multinucleate asexual spores"
temp$SporeType[which(temp$SporeName=="Zygospores")]<-"Multinucleate sexual spores"
temp$SporeType[which(temp$SporeName=="Teliospores")]<-"Multinucleate sexual spores"


# summarize data # My modification
sumbox<-temp %>%
  group_by(id, phylum,SporeType) %>%
  summarise(
    n=n(),
    meanV=mean(log_spore_volume),
    meanQ=mean(length_divided_by_width),
    sdV=sd(log_spore_volume),
    sdQ=sd(length_divided_by_width)
  ) %>%
  mutate(seV=sdV/sqrt(n)) %>%
  mutate(seQ=sdQ/sqrt(n))

#sumbox[sumbox$n==1,][,c("meanV","meanQ","sdV","sdQ","seV","seQ")]<-NA
#Orders for which we only have 1 species
just_one<-
  temp %>% 
  group_by(id) %>% 
  count() %>% 
  filter(n==1)
#These are the Basidiobolales and the Dimargitales

dropped.tree2<-drop.tip(dropped.tree2,dropped.tree2$tip.label[match(just_one$id, 
                                                                    dropped.tree2$tip.label)])

sumbox<-sumbox[-which(sumbox$n==1),]

#my changes:
grouped_tree <- groupClade(dropped.tree2, .node=c(72, 104,65), group_name = "Phyla")


# c("Ascomycota"          = "#001399", # This is blue
#   "Basidiomycota"       = "#C40233",#This is red 
#   "Glomeromycota"       = "#19461A",
#   "Zygomycota"          = "#19461A")#,This is greenish

#My modifications
pltre_neutral <- ggtree(grouped_tree, aes( color=Phyla))+ 
  geom_tiplab(size = 2)+ 
  theme_tree2()+
  ggplot2::xlim(0, 800)+
  scale_color_manual(
    values=c("black","#001399","#C40233","#19461A")
  ) +
  theme(legend.position = "none")

pltre_neutral
```

### Figure 1 Comparing spore size with that of seeds and eggs

```{r}
rbind(
  AllFungi%>%
        group_by(phylum,names_to_use,SporeName,Specific_sporeName)%>%
      summarise_at(c("spore_length","spore_width","SporeArea","SporeVolume","Q_ratio"),mean) %>% 
                          select(names_to_use,SporeVolume)%>%
                          rename(offsrpingSize=SporeVolume)%>%
                          #filter(good.names!="Glomus_tenue")%>%
                          mutate(Taxa="Fungi")%>%
                          mutate(offsrpingSize=offsrpingSize/((10000)^3)),#%>%
                          #mutate(Organism="Microorganism"),
                        
                        kewData%>%
                          select(species,value)%>%
                          rename(names_to_use=species, offsrpingSize=value)%>%
                          mutate(Taxa="Plants"),#%>%
                          #mutate(Organism="Macroorganism"),
                        BirdEgg%>%
                          select(Species,Volume)%>%
                          rename(names_to_use=Species,offsrpingSize=Volume)%>%
                          mutate(Taxa="Birds"))%>%
                      mutate(Taxa=factor(Taxa,levels =c("Birds","Plants","Fungi"))) %>% 
                      #mutate(Taxa=factor(Taxa,levels = c("Fungi","Plants","Birds"))) %>% #For the coord flip version is better in this reverse order
                          #mutate(Organism="Macroorganism"))%>%
                    
                    ggplot()+
                    aes(Taxa,offsrpingSize)+
                    geom_jitter(alpha=0.6,size=4,width = 0.3,aes(color=Taxa))+
                    geom_violin(alpha=0.5,
                          lwd=1,
                          fill=NA)+
                    #facet_grid(. ~ Organism, scales = "free")+
                    scale_y_log10(labels = scales::trans_format("log10", scales::math_format(10^.x)))+
                    labs(x="Offspring structure",y=expression("Offspring size as volume ("*c*"m³)"))+
                    scale_color_brewer(palette="Accent")+
                    scale_x_discrete(labels=c("Birds" = "Birds\n(eggs)", "Plants"="Plants\n(seeds)","Fungi"="Fungi\n(spores)"))+
                     my_theme2_5+
  coord_flip()

```

### Figure 1 Heatmaps

```{r}
# get list of taxa in order in which they occur on tree
d = fortify(dropped.tree2)
d = subset(d, isTip)
with(d, label[order(y, decreasing=T)])

# reorder id factor
sumbox$id<-factor(sumbox$id, levels = with(d, label[order(y, decreasing=F)]))


datos<-sumbox[order(factor(sumbox$id,levels=levels(sumbox$id))),]

datos$Vol_CV<-datos$sdV/datos$meanV

datos$Q_CV<-datos$sdQ/(datos$meanQ+1)#Adding 1 is a quick solution I am not sure about it

#rownames(datos)<-datos$id
datos<-pivot_wider(datos[,c("id","SporeType","meanV","meanQ","Vol_CV","Q_CV")],
                   names_from = SporeType,values_from = c(meanV,Vol_CV,meanQ,Q_CV))


datos1<-as.matrix(datos[,c( "meanV_Meiospores",#"sdV_Meiospores",
                            "meanV_Mitospores",#"sdV_Mitospores",
                            "meanV_Multinucleate sexual spores",
                            "meanV_Multinucleate asexual spores")])#"sdV_Multinucleate asexual spores",
                            #,"sdV_Multinucleate sexual spores"#,
                                                        #)])
rownames(datos1)<-datos$id


datos2<-as.matrix(datos[,c( "Vol_CV_Meiospores",
                            "Vol_CV_Mitospores",
                            "Vol_CV_Multinucleate sexual spores",
                            "Vol_CV_Multinucleate asexual spores")])
rownames(datos2)<-datos$id
##
datos3<-as.matrix(datos[,c( "meanQ_Meiospores",
                            "meanQ_Mitospores",
                            "meanQ_Multinucleate sexual spores",
                            "meanQ_Multinucleate asexual spores")])
rownames(datos3)<-datos$id

datos4<-as.matrix(datos[,c( "Q_CV_Meiospores",
                            "Q_CV_Mitospores",
                            "Q_CV_Multinucleate sexual spores",
                            "Q_CV_Multinucleate asexual spores")])
rownames(datos4)<-datos$id
```


```{r}
library(RColorBrewer)
coul<-colorRampPalette(brewer.pal(9, "Blues"))(100)
#Volume
heatmap(datos1,Colv = NA, Rowv = NA,cexCol = 0.5,cexRow = 0.5,scale = "none",
        col=hcl.colors(12, "YlOrRd", rev = TRUE))
legend(x="bottomright", legend=c("a", "b", "c","d","e"), 
       fill=hcl.colors(5, "YlOrRd", rev = TRUE)[5:1])

heatmap(datos2,Colv = NA, Rowv = NA,cexCol = 0.5,cexRow = 0.5,scale = "none",
        col=coul)
```


```{r}
#Aspect ratio
heatmap(datos3,Colv = NA, Rowv = NA,cexCol = 0.5,cexRow = 0.5,scale = "none",
        col=hcl.colors(12, "Greens", rev = TRUE))
legend(x="bottomright", legend=c("a", "b", "c","d","e"), 
       fill=hcl.colors(5, "Greens", rev = TRUE)[5:1])

heatmap(datos4,Colv = NA, Rowv = NA,cexCol = 0.5,cexRow = 0.5,scale = "none",
        col=coul)
legend(x="bottomright", legend=c("a", "b", "c","d","e"), 
       fill=colorRampPalette(brewer.pal(9, "Blues"))(5)[5:1])
```


## Figure 2 (colors were modified manually in inkscape)

```{r}
library(ggridges)
To_Analysis_c<-To_Analysis
To_Analysis_c$phylum[To_Analysis_c$phylum=="\"Zygomycetous fungi\""]<-"Zygomycetous fungi"

To_Analysis_c<-To_Analysis_c %>% 
  mutate(simpleFunct=factor(simpleFunct,levels = c("Saprotroph","Human","Plant Endophyte","Plant necrotroph","Plant Pathogen undefined","Plant Ectomycorrhizal","Insect","Lichen","Plant Pathogen Smut","Plant Pathogen Mildew","Plant Pathogen Rust")))
```

Figure 2 Obtaining the mean meiospores size for saprotrophic fungi
```{r}
tapply(log10(To_Analysis_c$SporeVolume[which(To_Analysis_c$SporeType=="Meiospores")]),
        list(To_Analysis_c$simpleFunct[which(To_Analysis_c$SporeType=="Meiospores")],
             To_Analysis_c$phylum[which(To_Analysis_c$SporeType=="Meiospores")]),
       mean)
```


Figure 2 Obtaining the mean mitospore size for saprotrophic fungi
```{r}
tapply(log10(To_Analysis_c$SporeVolume[which(To_Analysis_c$SporeType=="Meiospores")]),
        list(To_Analysis_c$simpleFunct[which(To_Analysis_c$SporeType=="Meiospores")],
             To_Analysis_c$phylum[which(To_Analysis_c$SporeType=="Meiospores")]),
       length)
```



Figure 2 Upper panel
```{r}
To_Analysis_c %>% 
  #filter(SporeType=="Meiospores"|SporeType=="Mitospores") %>% 
  filter(SporeType=="Meiospores") %>% 
  filter(!(SporeName=="Ascospores"&simpleFunct=="Plant Ectomycorrhizal")) %>% 
  filter(!(SporeName=="Ascospores"&simpleFunct=="Plant Pathogen Smut")) %>% 
  filter(!(phylum=="Basidiomycota"&simpleFunct=="Human")) %>% 
  mutate(simpleFunct=factor(simpleFunct,levels = c("Saprotroph","Human","Plant Endophyte","Plant necrotroph","Plant Pathogen undefined","Plant Ectomycorrhizal","Insect","Lichen","Plant Pathogen Smut","Plant Pathogen Mildew","Plant Pathogen Rust"))) %>%
  
  ggplot()+
  aes(y=simpleFunct,
      x=SporeVolume,
      
  )+
geom_density_ridges(aes(fill = phylum,point_alpha=0.05),jittered_points = TRUE, #position = "raincloud",
     alpha=0.9,scale = 2,rel_min_height = 0.01)+
  scale_x_log10(breaks=c(10^0,10^1,10^2,10^3,10^4,10^5,10^6,10^7),
                labels = scales::trans_format("log10", scales::math_format(10^.x)))+
  facet_wrap(.~phylum, strip.position =  "top",nrow = 1,drop=FALSE)+
  
    theme(title = element_text(size = 18),
        panel.background = element_blank(),
        
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line(size = 0.25, linetype = 'longdash',
                                        colour = "gray"),
        axis.title.y=element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(size = 10),#,angle = 45,hjust = 1),
        axis.text.y = element_text(size = 5),
        strip.text.x = element_text(size = 5),
        strip.text.y = element_blank(),
        legend.position = "none",
        panel.spacing = unit(-3, 'inches'))+
#   #For ascomycota
   geom_vline(aes(xintercept=10^2.64),color="red",linetype="longdash")+
 
#   #For basidiomycota
   geom_vline(aes(xintercept=10^2.06),color="blue",linetype="longdash")
 

```


Figure 2 Lower panel
```{r}

To_Analysis_c %>% 
  #filter(SporeType=="Meiospores"|SporeType=="Mitospores") %>% 
  filter(SporeType=="Mitospores") %>% 
  
  #filter(!(simpleFunct=="Plant Ectomycorrhizal")) %>% 
  filter(!(phylum=="Ascomycota"&simpleFunct=="Plant Pathogen Smut")) %>% #There is only 1 and it is actually a "false smut"
  filter(!(phylum=="Basidiomycota"&simpleFunct=="Human")) %>% #Thre is only 1
  filter(!(phylum=="Basidiomycota"&simpleFunct=="Plant Ectomycorrhizal")) %>% #There are only 3
  filter(!(phylum=="Basidiomycota"&simpleFunct=="Plant Pathogen undefined")) %>% #There are only 3
  
  #filter(phylum=="Ascomycota") %>% 
  #filter(phylum=="Basidiomycota") %>% 
  ggplot()+
  aes(y=simpleFunct,
      x=SporeVolume,
      
  )+
geom_density_ridges(aes(fill = phylum,point_alpha=0.05),jittered_points = TRUE, #position = "raincloud",
     alpha=0.9,scale = 2,rel_min_height = 0.01)+
  scale_x_log10(breaks=c(10^0,10^1,10^2,10^3,10^4,10^5,10^6,10^7),
                labels = scales::trans_format("log10", scales::math_format(10^.x)))+
  facet_wrap(.~phylum, strip.position =  "top",nrow = 1,drop=FALSE)+
  
    theme(title = element_text(size = 18),
        panel.background = element_blank(),
        
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line(size = 0.25, linetype = 'longdash',
                                        colour = "gray"),
        axis.title.y=element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(size = 10),#,angle = 45,hjust = 1),
        axis.text.y = element_text(size = 5),
        strip.text.x = element_text(size = 5),
        strip.text.y = element_blank(),
        legend.position = "none",
        panel.spacing = unit(-3, 'inches'))+
#   #For ascomycota
   geom_vline(aes(xintercept=10^2.17),color="red",linetype="longdash")+
 
#   #For basidiomycota
   geom_vline(aes(xintercept=10^2.6),color="green",linetype="longdash")+
 
# #For zygomycota
   geom_vline(aes(xintercept=10^2.44),color="blue",linetype="longdash")
```


## Figure S1
```{r}
data.frame(table(AllFungi[!duplicated(AllFungi$names_to_use),c("phylum")]))
  
Ascos<-c(`Spore` = 17656,`No_spore` = 66169)
Basidios<-c(`Spore` = 7530,`No_spore` = 40875)
Chytrids<-c(`Spore` = 2,`No_spore` = 1216)
Glomeros<-c(`Spore` = 274,`No_spore` = 10)
Zygos<-c(`Spore` = 295,`No_spore` = 1071)

library(waffle)
library(RColorBrewer)

waffle(
  Ascos / 100, rows = 20, size = 0.5,
  colors = c("blue","light blue"),
  title = "Ascomycota",
  #xlab = "1 square = 100 Species",
  pad = 3,
  legend_pos = "none"
  #use_glyph = "star"
)->A

waffle(
  Basidios / 100, rows = 12, size = 0.5,
  colors = c("blue","light blue"),
  title = "Basidiomycota",
  #xlab = "1 square = 100 Species",
  pad = 3,
  legend_pos = "none"
  #use_glyph = "star"
)->B

waffle(
  Chytrids , rows = 20, size = 0.5,
  colors = c("blue","light blue"),
  title = "Chytridiomycota",
  #xlab = "1 square = 1 Species", 
  pad = 3,
  legend_pos = "none"
  #use_glyph = "star"
)->C

waffle(
  Glomeros , rows = 7, size = 0.5,
  colors = c("blue","light blue"),
  title = "Glomeromycota",
  #xlab = "1 square = 1 Species",
  pad = 3,
  legend_pos = "none"
  #use_glyph = "star"
)->G

waffle(
  Zygos , rows = 20, size = 0.5,
  colors = c("blue","light blue"),
  title = "Zygomycota",
  #xlab = "1 square = 1 Species", 
  pad = 3,
  legend_pos = "none"
  #use_glyph = "star"
)->Z

#Ploting them. First the "higher fungi (A and B) and then the lower fungi (C,G,Z)
iron(A, B)
iron(C,G,Z)
```

## Figure S2
```{r}
Spore_functions%>%
  filter(!duplicated(names_to_use)) %>% 
  filter(!is.na(Life_style)) %>%
  filter(!grepl("-",host))%>%
  filter(!grepl("Fungi",host))%>%
  group_by(phylum,Guild_1)%>%
  count(phylum) %>% 
  ggplot()+
  aes(fill=Guild_1,x=phylum,y=n)+
  geom_bar(stat = "identity",position = "stack")+
  scale_color_brewer(palette="Set1")+
  labs(y="Number of species",x="Taxonomic group")+
  theme(title = element_text(size = 20),
        axis.title.x=element_blank(),
        axis.text.x = element_text(size = 20,angle = 45,hjust = 1),
        axis.text.y = element_text(size = 20),
        strip.text.x = element_text(size = 20),
        legend.text =  element_text(size = 15))+
  scale_fill_discrete(name="Functional groups")
```


## Figure S3

Ascomycota and Basidiomycota
```{r}

To_Analysis %>% 
  ungroup() %>% 
  filter(phylum=="Ascomycota" | phylum=="Basidiomycota") %>% 
  filter(!SporeName=="Teliospores") %>% 
  mutate(SporeName=gsub("Ascospores|Basidiospores","Sexual spores",SporeName)) %>% 
  
  ggplot()+
  aes(simpleFunct,SporeVolume,fill=Life_style)+
  geom_jitter(size=1.5, width = 0.3,alpha=0.8)+
  geom_violin(alpha=0.8, draw_quantiles=c(0.25, 0.5, 0.75))+
  #facet_grid(SporeName~phylum, scales = "free")+
  facet_grid(phylum~SporeName, scales = "free_y")+
  scale_color_brewer(palette="Set1")+
  scale_y_log10(labels = scales::trans_format("log10", scales::math_format(10^.x)))+
  labs(y=expression("Spore size as volume ("*mu*"m³)"))+
  scale_x_discrete(labels=c("AFree living" = "Free living", "Human"="Human pathogen","Insect"="Insect pathogen",
                            "Plant Endophyte"="Endophyte","Lichen" = "Lichen*","Plant Pathogen"="Plant Pathogen",
                            "Plant Ectomycorrhizal"="Ectomycorrhiza","Plant AMF"="Arbuscular Mycorrhiza"))+
  my_theme4+#theme(axis.text.x = element_blank(),axis.title.x=element_blank())+
  coord_flip()
```

Glomeromycota and "zygomycetous fungi"
```{r}
To_Analysis %>% 
  filter(!c(SporeName=="Azygospores" & phylum=="Zygomycota")) %>% 
  filter(phylum=="Zygomycota" | phylum=="Glomeromycota") %>% 
  mutate(phylum_="Basal clades") %>% 
  
  ggplot()+
  aes(simpleFunct,SporeArea,fill=Life_style)+
  geom_jitter(size=1.5, width = 0.3,alpha=0.8)+
  geom_violin(alpha=0.8, draw_quantiles=c(0.25, 0.5, 0.75))+
  facet_grid(phylum_~SporeName , scales = "free_y")+
  scale_color_brewer(palette="Set1")+
  scale_y_log10(labels = scales::trans_format("log10", scales::math_format(10^.x)))+
  labs(y=expression("Spore size as volume ("*mu*"m³)"))+
  scale_x_discrete(labels=c("AFree living" = "Free living", "Human"="Human pathogen","Insect"="Insect pathogen",
                            "Plant Endophyte"="Endophyte","Lichen" = "Lichen*","Plant Pathogen"="Plant Pathogen",
                            "Plant Ectomycorrhizal"="Ectomycorrhiza","Plant AMF"="Arbuscular Mycorrhiza"))+
  my_theme4+
  coord_flip()
```



## Standard major regression of spore length and spore width per spore type

```{r}
#Ascospores
model_ascospore_shape<-sma(spore_length ~ spore_width* simpleFunct, #multcomp = T,
                           data = Ascospores,log = "xy",slope.test =  1,elev.test = 0)

#Because the slope is never significantly different from 1 I change the formula to +:
model_ascospore_shape<-sma(spore_length ~ spore_width+ simpleFunct, #multcomp = T,
                           data = Ascospores,log = "xy",elev.test = 0,slope.test=1,robust = T)

summary(model_ascospore_shape)
```


```{r}
model_ascospore_shape$groupsummary[,c("group","Int","Elev_test_p","Slope","Slope_test_p")]
```

```{r}
plot(model_ascospore_shape)
```



# Conidia

```{r}
model_conidia_shape<-sma(spore_length ~ spore_width*simpleFunct,
                         data = Conidia,log = "xy",slope.test = 1,elev.test = 0,robust = T)


summary(model_conidia_shape)
```

```{r}
#model_conidia_shape$groupsummary[,c("group","Int","Elev_test_p")]
#model_conidia_shape$groupsummary[,c("group","Slope","Slope_test_p")]

model_conidia_shape$groupsummary[,c("group","Int","Elev_test_p","Slope","Slope_test_p")]
```


```{r}
plot(model_conidia_shape)
```


```{r}
model_basidiospore_shape<-sma(spore_length ~ spore_width*simpleFunct,#multcomp = T,
                           data = Basidiospores,log = "xy",slope.test = 1,elev.test = 0)

#Because the slope is in few cases significantly different from 1 I change the formula to +:
model_basidiospore_shape<-sma(spore_length ~ spore_width+simpleFunct,#multcomp = T,
                           data = Basidiospores,log = "xy",elev.test = 0, slope.test = 1 ,robust = T)


summary(model_basidiospore_shape)
```
```{r}
model_basidiospore_shape$groupsummary[,c("group","Int","Elev_test_p","Slope","Slope_test_p")]
```

```{r}
plot(model_basidiospore_shape)
```

```{r}

shape_summary<-rbind(
cbind(LogQ=as.vector(tapply(log10(Ascospores$Q_ratio),Ascospores$simpleFunct,mean)),model_ascospore_shape$groupsummary[,c("group","Int","Int_lowCI","Int_highCI","Elev_test_p","Slope","Slope_test_p","Slope_lowCI","Slope_highCI")],SporeName=rep("Ascospores",8)),
cbind(LogQ=as.vector(tapply(log10(Conidia$Q_ratio),Conidia$simpleFunct,mean)),model_conidia_shape$groupsummary[,c("group","Int","Int_lowCI","Int_highCI","Elev_test_p","Slope","Slope_test_p","Slope_lowCI","Slope_highCI")],SporeName=rep("Conidia",8)),
cbind(LogQ=as.vector(tapply(log10(Basidiospores$Q_ratio),Basidiospores$simpleFunct,mean)),model_basidiospore_shape$groupsummary[,c("group","Int","Int_lowCI","Int_highCI","Elev_test_p","Slope","Slope_test_p","Slope_lowCI","Slope_highCI")],SporeName=rep("Basidiospores",8))
)

shape_summary

```



```{r,fig.height = 15, fig.width = 8}
shape_summary %>% 
  #filter(SporeName!="Basidiospores") %>% 
  ggplot()+
  aes(x=LogQ,y=Int)+
  geom_point()+
  facet_wrap(.~SporeName)


shape_summary %>% 
  filter(group!="Plant Pathogen Smut") %>% 
  mutate(group_name=factor(group,levels = c("Saprotroph","Human","Plant Endophyte","Plant necrotroph",
                                            "Plant Pathogen undefined","Insect","Lichen","Plant Pathogen Mildew","Plant Pathogen Rust","Plant Ectomycorrhizal"))) %>% 
  ggplot()+
  aes(x=group_name,y=Int,ymin=Int_lowCI,ymax=Int_highCI)+
  geom_pointrange(aes(col=group))+
  geom_point(aes(y=LogQ,col=group),shape=17,size=2.5)+
  facet_wrap(.~SporeName,nrow = 3,scales = "free_y")+
  my_theme2_5+
  labs(y="Spheroidicity")+
  #scale_y_continuous(minor_breaks = c(-1.5,1.5,0.5))+
  coord_flip()

#check out annotation ticks for log
```


```{r,fig.height = 15, fig.width = 8}
shape_summary %>% 
  filter(group!="Plant Pathogen Smut") %>% 
  mutate(group_name=factor(group,levels = c("Saprotroph","Human","Plant Endophyte","Plant necrotroph",
                                            "Plant Pathogen undefined","Insect","Lichen","Plant Pathogen Mildew","Plant Pathogen Rust","Plant Ectomycorrhizal"))) %>% 
  ggplot()+
  aes(x=group_name,y=Slope,ymin=Slope_lowCI,ymax=Slope_highCI)+
  geom_pointrange(aes(col=group))+
  facet_wrap(.~SporeName,nrow = 3,scales = "free_y")+
  my_theme2_5+
  labs(y="Shape to Size allometric exponent")+
  #scale_y_continuous(minor_breaks = c(-1.5,1.5,0.5))+
  coord_flip()
```


