---
title: "What are the questions -- with variation partitioning"
output: word_document
---


```{r, message=FALSE, warning=FALSE, include=FALSE}
library(smatr)
library(performance)
library(tidyverse)
library(lme4)
source("MatchingSpore_FunctionData.R")
```

1. How much phylogeny explain variation in spore size?


In my opinion we should be using all the data (~60K entries) and taxonomy as a random factor (only using spore type as fixed effect). **Note this gives different results when I run it compared with the output file Carlos sent.**

```{r, echo=FALSE}

model1<-
lmer(log10(SporeVolume)~SporeType+(1|phylum/class/order/family),data = 
       To_Analysis)

```


```{r}
r2_nakagawa(model1)
```

This result indicates that spore type and taxonmy explain up to 54% of the variation in the total dataset. Spore type explain alone explain very little (7%), indictating that these two variables are highly correlated (particularly at kingdom-wide level).
Thus, the conclusion is is that indeed, taxonomy is a very important “driver” and we should take it into account.

```{r, message=FALSE, echo=FALSE}
# conidia_lifestyle<-
#   lmer(log10(SporeVolume)~simpleFunct+(1|order),data = 
#        Conidia)
# dat <- read_csv('output/df_species_byPrimerSet.csv')
dat <- read_csv('output/df_species_noPrimers.csv')

dat$names_to_use<-gsub("_"," ",dat$Species)

# trial<-left_join(dat %>% 
#                   select(names_to_use,Primers.name,host.assoc, ends_with('mean')), 
#                  To_Analysis[,c("names_to_use","SporeType","simpleFunct","SporeVolume", "Q_ratio",
#                                 "phylum","class","order")])
trial<-left_join(dat %>% 
                  select(names_to_use, host.assoc, log10.sporeVolume, log10.sporeAspectRatio, ends_with('mean'), n_samples_env), 
                 To_Analysis[,c("names_to_use","SporeType","simpleFunct","Life_style",
                                "phylum","class","order")])

# this is weird that there are species without assignments
trial<-trial[complete.cases(trial),] %>% 
  filter(!order == 'Not assigned') %>% 
  mutate(Life_style = recode(Life_style, Plant = "Facultative association"), 
         Life_style = ordered(Life_style))#, 
         # Life_style = poly(as.numeric(Life_style, 1)))

# table(trial$simpleFunct)
# filter(trial, simpleFunct == 'Human')

# libraries for tables
library(broom)
library(pixiedust)

```

2. So, once taxonomy is taken into account, how important is life-style? Relative to climatic conditions?

For this we subset the data to: a) include only the species for which we have life-style data, and b) we got climatic varialbes. This reduces the original dataset from 60K entries to `r nrow(trial)` entries. **Note that I had to remove some rows in which taxonomy was 'Not assigned', which shouldn't be the case.**

## Spore volume
```{r, echo=FALSE}

### spore volume

## global model
m1 <- lm(log10.sporeVolume ~ order + Life_style + SporeType + 
             mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# # plot to check for nonlinearities
# tmp <- m1$model %>% 
#   mutate_if(is.character, as.factor)
# for(i in 2:ncol(tmp)) plot(tmp$log10.sporeVolume ~ tmp[, i], main=names(tmp)[i]) 

## results for individual variables
m1.var <- vector('list', length=ncol(m1$model))
names(m1.var) <- c('<none>', names(m1$model)[2:ncol(m1$model)])
m1.var[[1]] <- m1$model
for(i in 2:length(m1.var)) m1.var[[i]] <- m1$model[, -i]
m1.aic <- lapply(m1.var, function(x)AIC(m1, lm(log10.sporeVolume ~ ., data=x)))
m1.drp <- data.frame(term = names(m1.var), 
                     Df = sapply(m1.aic, function(x)x[2, 'df']), 
                     AIC = sapply(m1.aic, function(x)x[2, 'AIC']), 
                     R2 = sapply(m1.var, function(x)summary(lm(log10.sporeVolume ~ ., data=x))[['r.squared']])) %>% 
  mutate(type = c('global', rep('subset', n() - 1)), 
         dAIC = AIC - AIC[1], 
         dR2 = R2 - R2[1]) %>% 
  group_by(type) %>% 
  arrange(desc(dAIC), .by_group=TRUE) %>% 
  ungroup() %>% 
  select(term, Df, AIC, dAIC, R2, dR2)

## results for variable groups
# drop taxonomy
m2.tax <- lm(log10.sporeVolume ~ Life_style + SporeType + 
               mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop function
m2.fun <- lm(log10.sporeVolume ~ order + SporeType + 
               mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop spore type
m2.typ <- lm(log10.sporeVolume ~ order + Life_style + 
               mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop environment
m2.env <- lm(log10.sporeVolume ~ order + Life_style + SporeType, data=trial)
# make table
m2.drp <- data.frame(term=c('<none>', 'taxonomy', 'Life_style', 'SporeType', 'climate'), 
                     df=AIC(m1, m2.tax, m2.fun, m2.typ, m2.env)[['df']], 
                     AIC=AIC(m1, m2.tax, m2.fun, m2.typ, m2.env)[['AIC']], 
                     R2=sapply(list(m1, m2.tax, m2.fun, m2.typ, m2.env), function(x)summary(x)[['r.squared']]), 
                     row.names=NULL) %>% 
  mutate(type = c('<none>', rep('subset', n() - 1)), 
         dAIC = AIC - AIC[1], 
         dR2 = R2 - R2[1]) %>% 
  group_by(type) %>% 
  arrange(desc(dAIC), .by_group=TRUE) %>% 
  ungroup() %>% 
  select(term, df, AIC, dAIC, R2, dR2)

# Table 1
dust(m1.drp, caption='Table X. Importance of individual variables in linear models for explaining variation among species in spore volume. Degrees of freedom (df), Akaike\'s Information Criterion (AIC), delta AIC (dAIC), R^2 (R2) and delta R^2 (dR2) were calculated by dropping terms from a global model containing all terms (indicated in italics).') %>%
  sprinkle_print_method('markdown')  %>%
  sprinkle(rows=1, italic=TRUE) %>%
  sprinkle(cols=c(3, 4), round=1) %>%
  sprinkle(cols=c(5, 6), round=3)

# Table 2
dust(m2.drp, caption='Table X. Importance of variable groupss in linear models for explaining variation among species in spore volume. Degrees of freedom (df), Akaike\'s Information Criterion (AIC), delta AIC (dAIC), R^2 (R2) and delta R^2 (dR2) were calculated by dropping groups of terms from a global model containing all terms (indicated in italics). ') %>%
  sprinkle_print_method('markdown')  %>%
  sprinkle(rows=1, italic=TRUE) %>%
  sprinkle(cols=c(3, 4), round=1) %>%
  sprinkle(cols=c(5, 6), round=3)
  
```

All these outputs confirm the same trend: Taxonomy is important. In fact, using it as fixed factor, shows that is the single most important factor (removing makes the model really bad, adding improves a lot)

Next in “importance” is lifestyle (removing makes the model bad, adding lifestyle makes it better). More so than any other climatic variable or spore type, although the explanatory power of the model is pretty small (< 1\%). **Maybe we should use R2 instead (or in addition to) adjusted R2**
The pattern also holds even when we simplify lifestyles to be either host associated or freeliving fungi. **Not included here**


## Spore aspect ratio
```{r, echo=FALSE}

### spore aspect ratio

## global model
m1 <- lm(log10.sporeAspectRatio ~ order + Life_style + SporeType + 
             mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# # plot to check for nonlinearities
# tmp <- m1$model %>% 
#   mutate_if(is.character, as.factor)
# for(i in 2:ncol(tmp)) plot(tmp$log10.sporeAspectRatio ~ tmp[, i], main=names(tmp)[i]) 

## results for individual variables
m1.var <- vector('list', length=ncol(m1$model))
names(m1.var) <- c('<none>', names(m1$model)[2:ncol(m1$model)])
m1.var[[1]] <- m1$model
for(i in 2:length(m1.var)) m1.var[[i]] <- m1$model[, -i]
m1.aic <- lapply(m1.var, function(x)AIC(m1, lm(log10.sporeAspectRatio ~ ., data=x)))
m1.drp <- data.frame(term = names(m1.var), 
                     Df = sapply(m1.aic, function(x)x[2, 'df']), 
                     AIC = sapply(m1.aic, function(x)x[2, 'AIC']), 
                     R2 = sapply(m1.var, function(x)summary(lm(log10.sporeAspectRatio ~ ., data=x))[['r.squared']])) %>% 
  mutate(type = c('global', rep('subset', n() - 1)), 
         dAIC = AIC - AIC[1], 
         dR2 = R2 - R2[1]) %>% 
  group_by(type) %>% 
  arrange(desc(dAIC), .by_group=TRUE) %>% 
  ungroup() %>% 
  select(term, Df, AIC, dAIC, R2, dR2)

## results for variable groups
# drop taxonomy
m2.tax <- lm(log10.sporeAspectRatio ~ Life_style + SporeType + 
               mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop function
m2.fun <- lm(log10.sporeAspectRatio ~ order + SporeType + 
               mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop spore type
m2.typ <- lm(log10.sporeAspectRatio ~ order + Life_style + 
               mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop environment
m2.env <- lm(log10.sporeAspectRatio ~ order + Life_style + SporeType, data=trial)
# make table
m2.drp <- data.frame(term=c('<none>', 'taxonomy', 'Life_style', 'SporeType', 'climate'), 
                     df=AIC(m1, m2.tax, m2.fun, m2.typ, m2.env)[['df']], 
                     AIC=AIC(m1, m2.tax, m2.fun, m2.typ, m2.env)[['AIC']], 
                     R2=sapply(list(m1, m2.tax, m2.fun, m2.typ, m2.env), function(x)summary(x)[['r.squared']]), 
                     row.names=NULL) %>% 
  mutate(type = c('<none>', rep('subset', n() - 1)), 
         dAIC = AIC - AIC[1], 
         dR2 = R2 - R2[1]) %>% 
  group_by(type) %>% 
  arrange(desc(dAIC), .by_group=TRUE) %>% 
  ungroup() %>% 
  select(term, df, AIC, dAIC, R2, dR2)

# Table 1
dust(m1.drp, caption='Table X. Importance of individual variables in linear models for explaining variation among species in spore shape Degrees of freedom (df), Akaike\'s Information Criterion (AIC), delta AIC (dAIC), R^2 (R2) and delta R^2 (dR2) were calculated by dropping terms from a global model containing all terms (indicated in italics).') %>%
  sprinkle_print_method('markdown')  %>%
  sprinkle(rows=1, italic=TRUE) %>%
  sprinkle(cols=c(3, 4), round=1) %>%
  sprinkle(cols=c(5, 6), round=3)

# Table 2
dust(m2.drp, caption='Table X. Importance of variable groupss in linear models for explaining variation among species in spore shape Degrees of freedom (df), Akaike\'s Information Criterion (AIC), delta AIC (dAIC), R^2 (R2) and delta R^2 (dR2) were calculated by dropping groups of terms from a global model containing all terms (indicated in italics). ') %>%
  sprinkle_print_method('markdown')  %>%
  sprinkle(rows=1, italic=TRUE) %>%
  sprinkle(cols=c(3, 4), round=1) %>%
  sprinkle(cols=c(5, 6), round=3)
  
```

Spore shape shows a similar pattern with regard to taxonomy being the most important, with spore type being second most importance. Climate and lifestyle ends up being of similar importance. The most important climate variable is solar radiation, with species with more ovoid spores being found in environments with higher solar radiation.


3. How much life-style influence the expected spore size ~ geographic extent?
This is another story :)


```{r, echo=FALSE, eval=FALSE}

##### analysis of extent

temp<-left_join(read_csv('output/df_species_byPrimerSet.csv') %>%
                   mutate(names_to_use = gsub("_", " ", Species)) %>%
                   select(names_to_use, host.assoc, Primers.name, log10.sporeVolume, log10.sporeAspectRatio, starts_with('geo_'), ends_with('_sd'), n_samples_env),
                 To_Analysis[,c("names_to_use","SporeType","simpleFunct",
                                "phylum","class","order")])

ggplot(temp %>% filter(!is.na(host.assoc)), aes(x=log10.sporeVolume, y=car::logit(geo_extent_prop), colour=host.assoc)) + 
  geom_point(alpha=0.5, shape=16) + 
  # stat_smooth(method='lm', formula=y ~ poly(x, 2), alpha=0.5) +
  stat_smooth(method='lm', alpha=0.5) #+ 
  # facet_grid(cols=vars(phylum))

# fit linear model
m1 <- lme4::lmer(car::logit(geo_extent_prop) ~ log10.sporeVolume * host.assoc +
                   (1+log10.sporeVolume|order) + (1|Primers.name) + (1|SporeType), data=temp)
summary(m1)
car::Anova(m1)
# car::qqPlot(resid(m1))
# plot(m1)
# anova(m1, 
#       lme4::lmer(car::logit(geo_extent_prop) ~ log10.sporeVolume * host.assoc * SporeType +
#                    (1|order) + (1|Primers.name), data=temp))
# anova(m1, 
#       lme4::lmer(car::logit(geo_extent_prop) ~ log10.sporeVolume * host.assoc * SporeType +
#                    (1+log10.sporeVolume|order), data=temp))

```

```{r}

temp<-left_join(read_csv('output/df_species_byPrimerSet.csv') %>%
                   mutate(names_to_use = gsub("_", " ", Species)) %>%
                   select(names_to_use, host.assoc, Primers.name, log10.sporeVolume, log10.sporeAspectRatio, starts_with('geo_'), ends_with('_sd'), n_samples_env),
                 To_Analysis[,c("names_to_use","SporeType","simpleFunct","Life_style",
                                "phylum","class","order")]) %>% 
  filter(!order == 'Not assigned') %>% 
  mutate(Life_style = recode(Life_style, Plant = "Facultative association"), 
         Life_style = ordered(Life_style))#, 
         # Life_style = poly(as.numeric(Life_style, 1)))

```


## Spore volume

```{r}

m1.list <- list(
  # geographic extent -- maximum distance between samples
  dist=lmer(car::logit(geo_extent_prop) ~ log10.sporeVolume * Life_style * SporeType + 
                 (1|order) + (1|Primers.name), data=temp), 
  # geographic extent -- habitat area, 
  area=lmer(car::logit(geo_area_prop) ~ log10.sporeVolume * Life_style * SporeType + 
                 (1|order) + (1|Primers.name), data=temp), 
  # environmental extent -- MAT
  mat=lmer(sqrt(mat_sd) ~ log10.sporeVolume * Life_style * SporeType + 
                 (1|order) + (1|Primers.name), data=temp), 
  # environmental extent -- MAP
  map=lmer(sqrt(map_sd) ~ log10.sporeVolume * Life_style * SporeType + 
                 (1|order) + (1|Primers.name), data=temp), 
  # environmental extent -- solar radiation
  maxsrad=lmer(sqrt(maxsrad_sd) ~ log10.sporeVolume * Life_style * SporeType + 
                 (1|order) + (1|Primers.name), data=temp), 
  # environmental extent -- minimum vapour pressure
  minvapr=lmer(sqrt(minvapr_sd) ~ log10.sporeVolume * Life_style * SporeType + 
                 (1|order) + (1|Primers.name), data=temp), 
  # general extent -- number of samples
  nsamp=lmer(sqrt(n_samples_env) ~ log10.sporeVolume * Life_style * SporeType + 
                 (1|order) + (1|Primers.name), data=temp)
)
  
lapply(m1.list, function(x)car::Anova(x))
# m1.coef <- sapply(m1.list, function(x)coef(x)[['order']][1, c('log10.sporeVolume', 'host.assocyes', 'log10.sporeVolume:host.assocyes')])
# m1.pval <- sapply(m1.list, function(x)car::Anova(x)[c('log10.sporeVolume', 'host.assoc', 'log10.sporeVolume:host.assoc'), 'Pr(>Chisq)'])

```

## Spore aspect ratio

```{r}

m2.list <- list(
  # geographic extent -- maximum distance between samples
  dist=lmer(car::logit(geo_extent_prop) ~ log10.sporeAspectRatio * Life_style * SporeType + 
                 (1|order) + (1|Primers.name), data=temp), 
  # geographic extent -- habitat area, 
  area=lmer(car::logit(geo_area_prop) ~ log10.sporeAspectRatio * Life_style * SporeType + 
                 (1|order) + (1|Primers.name), data=temp), 
  # environmental extent -- MAT
  mat=lmer(sqrt(mat_sd) ~ log10.sporeAspectRatio * Life_style * SporeType + 
                 (1|order) + (1|Primers.name), data=temp), 
  # environmental extent -- MAP
  map=lmer(sqrt(map_sd) ~ log10.sporeAspectRatio * Life_style * SporeType + 
                 (1|order) + (1|Primers.name), data=temp), 
  # environmental extent -- solar radiation
  maxsrad=lmer(sqrt(maxsrad_sd) ~ log10.sporeAspectRatio * Life_style * SporeType + 
                 (1|order) + (1|Primers.name), data=temp), 
  # environmental extent -- minimum vapour pressure
  minvapr=lmer(sqrt(minvapr_sd) ~ log10.sporeAspectRatio * Life_style * SporeType + 
                 (1|order) + (1|Primers.name), data=temp), 
  # general extent -- number of samples
  nsamp=lmer(sqrt(n_samples_env) ~ log10.sporeAspectRatio * Life_style * SporeType + 
                 (1|order) + (1|Primers.name), data=temp)
)
  
lapply(m2.list, function(x)car::Anova(x))
# m2.coef <- sapply(m2.list, function(x)coef(x)[['order']][1, c('log10.sporeAspectRatio', 'host.assocyes', 'log10.sporeAspectRatio:host.assocyes')])
# m2.pval <- sapply(m2.list, function(x)car::Anova(x)[c('log10.sporeAspectRatio', 'host.assoc', 'log10.sporeAspectRatio:host.assoc'), 'Pr(>Chisq)'])

```

