---
title: "What are the questions -- with variation partitioning"
output: word_document
---


```{r, message=FALSE, warning=FALSE, include=FALSE}
library(smatr)
library(performance)
library(tidyverse)
library(lme4)
library(lmerTest)
source("MatchingSpore_FunctionData.R")
```

1. How much phylogeny explain variation in spore size?


In my opinion we should be using all the data (~60K entries) and taxonomy as a random factor (only using spore type as fixed effect). 

**Note this gives different results when I run it compared with the output file Carlos sent.**

```{r, echo=FALSE}

model1<-
lmer(log10(SporeVolume)~SporeType+(1|phylum/class/order/family),data = 
       To_Analysis)

r2_nakagawa(model1)
```
This result indicates that spore type and taxonomy explain up to 74% of the variation in the subset of the species for which we also have functiona data. Spore type alone explain roughly 8%, indictating that these two variables are highly correlated (particularly at kingdom-wide level).

**Note on the previous comment **

The difference between the results of the model above and what I send previously is because I was using all species in the dataset (that is the dataframe called "AllFungi" of ~67 K entries). The model above uses "To_Analysis" which only contains the subset for which we also have functional data

```{r}
model_allFungi<-
lmer(log10(SporeVolume)~SporeType+(1|phylum/class/order),data = 
       AllFungi[-which(is.na(AllFungi$SporeType)),])

r2_nakagawa(model_allFungi)
```
This result indicates that spore type and taxonmy explain up to 54% (so less than above) of the variation in the total dataset (regardless of whether we have functional data). In this case spore type stay more or less the same (7%).

In either case, the conclusion is is that indeed, taxonomy and then spore type are very important “driver” and we should take it into account.


2. So, once taxonomy is taken into account, how important is life-style? Relative to climatic conditions?



```{r, message=FALSE, echo=FALSE}
# conidia_lifestyle<-
#   lmer(log10(SporeVolume)~simpleFunct+(1|order),data = 
#        Conidia)
# dat <- read_csv('output/df_species_byPrimerSet.csv')
dat <- read_csv('output/df_species_noPrimers.csv')

dat$names_to_use<-gsub("_"," ",dat$Species)

# trial<-left_join(dat %>% 
#                   select(names_to_use,Primers.name,host.assoc, ends_with('mean')), 
#                  To_Analysis[,c("names_to_use","SporeType","simpleFunct","SporeVolume", "Q_ratio",
#                                 "phylum","class","order")])
trial<-left_join(dat %>% 
                  select(names_to_use, host.assoc, log10.sporeVolume, log10.sporeAspectRatio, ends_with('mean'), n_samples_env), 
                 To_Analysis[,c("names_to_use","SporeType","simpleFunct","Life_style",
                                "phylum","class","order")])

# this is weird that there are species without assignments
trial<-trial[complete.cases(trial),] %>% 
  filter(!order == 'Not assigned') %>% 
  mutate(Life_style = recode(Life_style, Plant = "Facultative association"), 
         Life_style = ordered(Life_style))#, 
         # Life_style = poly(as.numeric(Life_style, 1)))

# table(trial$simpleFunct)
# filter(trial, simpleFunct == 'Human')

# libraries for tables
library(broom)
library(pixiedust)

```


For this we subset the data to: a) include only the species for which we have life-style data, and b) we got climatic varialbes. This reduces the original dataset from 60K entries to `r nrow(trial)` entries. **Note that I had to remove some rows in which taxonomy was 'Not assigned', which shouldn't be the case.**

**I check and indeed we do not have taxonomy information for those species neither in from the Catalogue of Life nor from Mycobank** 


## Spore volume
```{r, echo=FALSE}

# library for Akaike weights
library(MuMIn)

### spore volume

## global model
m1 <- lm(log10.sporeVolume ~ order + Life_style + SporeType + 
             mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial, na.action=na.fail)
# # plot to check for nonlinearities
# tmp <- m1$model %>% 
#   mutate_if(is.character, as.factor)
# for(i in 2:ncol(tmp)) plot(tmp$log10.sporeVolume ~ tmp[, i], main=names(tmp)[i]) 

## results for individual variables
m1.var <- vector('list', length=ncol(m1$model))
names(m1.var) <- c('<none>', names(m1$model)[2:ncol(m1$model)])
m1.var[[1]] <- m1$model
for(i in 2:length(m1.var)) m1.var[[i]] <- m1$model[, -i]
m1.aic <- lapply(m1.var, function(x)AIC(m1, lm(log10.sporeVolume ~ ., data=x)))
m1.drp <- data.frame(term = names(m1.var), 
                     Df = sapply(m1.aic, function(x)x[2, 'df']), 
                     AIC = sapply(m1.aic, function(x)x[2, 'AIC']), 
                     R2 = sapply(m1.var, function(x)summary(lm(log10.sporeVolume ~ ., data=x))[['r.squared']])) %>% 
  mutate(type = c('global', rep('subset', n() - 1)), 
         dAIC = AIC - AIC[1], 
         dR2 = R2 - R2[1]) %>% 
  group_by(type) %>% 
  arrange(desc(dAIC), .by_group=TRUE) %>% 
  ungroup() %>% 
  select(term, Df, AIC, dAIC, R2, dR2)

## results using dredging
m1.drdg <- dredge(m1)
m1.drp <- left_join(m1.drp, data.frame(term = names(importance(m1.drdg)), 
                             sumWeights = importance(m1.drdg), row.names=NULL)) %>% 
  mutate(sumWeights = replace_na(sumWeights, '')) %>% 
  select(-R2, -dR2)

## results for variable groups
# drop taxonomy
m2.tax <- lm(log10.sporeVolume ~ Life_style + SporeType + 
               mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop function
m2.fun <- lm(log10.sporeVolume ~ order + SporeType + 
               mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop spore type
m2.typ <- lm(log10.sporeVolume ~ order + Life_style + 
               mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop environment
m2.env <- lm(log10.sporeVolume ~ order + Life_style + SporeType, data=trial, na.action=na.fail)
# make table
m2.drp <- data.frame(term=c('<none>', 'taxonomy', 'Life_style', 'SporeType', 'climate'), 
                     df=AIC(m1, m2.tax, m2.fun, m2.typ, m2.env)[['df']], 
                     AIC=AIC(m1, m2.tax, m2.fun, m2.typ, m2.env)[['AIC']], 
                     R2=sapply(list(m1, m2.tax, m2.fun, m2.typ, m2.env), function(x)summary(x)[['r.squared']]), 
                     row.names=NULL) %>% 
  mutate(type = c('<none>', rep('subset', n() - 1)), 
         dAIC = AIC - AIC[1], 
         dR2 = R2 - R2[1]) %>% 
  group_by(type) %>% 
  arrange(desc(dAIC), .by_group=TRUE) %>% 
  ungroup() %>% 
  select(term, df, AIC, dAIC, R2, dR2)

## results using dredging
m2.drdg <- list(m1, m2.tax, m2.fun, m2.typ, m2.env, 
                lm(log10.sporeVolume ~ SporeType + mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial),  
                lm(log10.sporeVolume ~ Life_style + mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial),  
                lm(log10.sporeVolume ~ Life_style + SporeType, data=trial), 
                lm(log10.sporeVolume ~ order + SporeType, data=trial),  
                lm(log10.sporeVolume ~ order + mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial),  
                lm(log10.sporeVolume ~ order + Life_style, data=trial),  
                lm(log10.sporeVolume ~ SporeType, data=trial), 
                lm(log10.sporeVolume ~ order, data=trial),  
                lm(log10.sporeVolume ~ mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial),  
                lm(log10.sporeVolume ~ Life_style, data=trial))
m2.drp <- left_join(m2.drp, data.frame(term = c('<none>', 'taxonomy', 'Life_style', 'SporeType', 'climate'), 
                             sumWeights = c(NA, importance(m2.drdg)[1:4]), row.names=NULL)) %>% 
  select(-R2, -dR2)

# Table 1
dust(m1.drp, caption='Table X. Importance of individual variables in linear models for explaining variation among species in spore volume. Degrees of freedom (df), Akaike\'s Information Criterion (AIC) and delta AIC (dAIC) were calculated by dropping terms from a global model containing all terms (indicated in italics). Akaike weights (sumWeights) were calculated from AIC scores following automated model selection using all terms.') %>%
  sprinkle_print_method('markdown')  %>%
  sprinkle(rows=1, italic=TRUE) %>%
  sprinkle(cols=c(3, 4), round=1) %>%
  sprinkle(cols=5, round=2)

# Table 2
dust(m2.drp, caption='Table X. Importance of variable groupss in linear models for explaining variation among species in spore volume. Degrees of freedom (df), Akaike\'s Information Criterion (AIC) and delta AIC (dAIC) were calculated by dropping groups of terms from a global model containing all terms (indicated in italics). Akaike weights (sumWeights) were calculated from AIC scores following automated model selection using all groups of terms.') %>%
  sprinkle_print_method('markdown')  %>%
  sprinkle(rows=1, italic=TRUE) %>%
  sprinkle(cols=c(3, 4), round=1) %>%
  sprinkle(cols=5, round=2)
  
```

All these outputs confirm the same trend: Taxonomy is important. In fact, using it as fixed factor, shows that is the single most important factor (removing makes the model really bad, adding improves a lot)

Next in “importance” is lifestyle (removing makes the model bad, adding lifestyle makes it better). More so than any other climatic variable or spore type, although the explanatory power of the model is pretty small (< 1\%). **Maybe we should use R2 instead (or in addition to) adjusted R2**
The pattern also holds even when we simplify lifestyles to be either host associated or freeliving fungi. **Not included here**


**Carlos approach using mixed effects models where taxonomy is a random factor**

Having taxonomy and spore type as random factors (not including multinucleate spores because they only include AMF which only inlcude one type of lifestyle plus 3 zygomycetous fungi) 

```{r, echo=FALSE,message=FALSE,warning=FALSE}
m_all <- lmer(log10.sporeVolume ~ Life_style + 
             mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean+
             (1|phylum/order)+(1|SporeType),  REML=TRUE,data=trial %>% filter(!grepl("Multi",SporeType)) %>% mutate_at(c("mat_mean", "map_mean", "seasonality_t_mean", "seasonality_p_mean", "maxsrad_mean","minvapr_mean"),scale))

m_no_life_style <- lmer(log10.sporeVolume ~ #Life_style + 
             mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean+
             (1|phylum/order)+(1|SporeType),  REML=TRUE,data=trial %>% filter(!grepl("Multi",SporeType)) %>% mutate_at(c("mat_mean", "map_mean", "seasonality_t_mean", "seasonality_p_mean", "maxsrad_mean","minvapr_mean"),scale))

m_no_mat_mean <- lmer(log10.sporeVolume ~ Life_style + 
             #mat_mean + 
               map_mean + 
               seasonality_t_mean + 
               seasonality_p_mean + 
               maxsrad_mean + 
               minvapr_mean+
             (1|phylum/order)+(1|SporeType),  REML=TRUE,data=trial %>% filter(!grepl("Multi",SporeType)) %>% mutate_at(c("mat_mean", "map_mean", "seasonality_t_mean", "seasonality_p_mean", "maxsrad_mean","minvapr_mean"),scale))


m_no_map_mean <- lmer(log10.sporeVolume ~ Life_style + 
               mat_mean + 
               #map_mean + 
               seasonality_t_mean + 
               seasonality_p_mean + 
               maxsrad_mean + 
               minvapr_mean+
             (1|phylum/order)+(1|SporeType),  REML=TRUE,data=trial %>% filter(!grepl("Multi",SporeType)) %>% mutate_at(c("mat_mean", "map_mean", "seasonality_t_mean", "seasonality_p_mean", "maxsrad_mean","minvapr_mean"),scale))

m_no_seasonality_t_mean <- lmer(log10.sporeVolume ~ Life_style + 
               mat_mean + 
               map_mean + 
               #seasonality_t_mean + 
               seasonality_p_mean + 
               maxsrad_mean + 
               minvapr_mean+
             (1|phylum/order)+(1|SporeType),  REML=TRUE,data=trial %>% filter(!grepl("Multi",SporeType)) %>% mutate_at(c("mat_mean", "map_mean", "seasonality_t_mean", "seasonality_p_mean", "maxsrad_mean","minvapr_mean"),scale))

m_no_seasonality_p_mean <- lmer(log10.sporeVolume ~ Life_style + 
               mat_mean + 
               map_mean + 
               seasonality_t_mean + 
               #seasonality_p_mean + 
               maxsrad_mean + 
               minvapr_mean+
             (1|phylum/order)+(1|SporeType),  REML=TRUE,data=trial %>% filter(!grepl("Multi",SporeType)) %>% mutate_at(c("mat_mean", "map_mean", "seasonality_t_mean", "seasonality_p_mean", "maxsrad_mean","minvapr_mean"),scale))

m_no_maxsrad_mean <- lmer(log10.sporeVolume ~ Life_style + 
               mat_mean + 
               map_mean + 
               seasonality_t_mean + 
               seasonality_p_mean + 
               #maxsrad_mean + 
               minvapr_mean+
             (1|phylum/order)+(1|SporeType),  REML=TRUE,data=trial %>% filter(!grepl("Multi",SporeType)) %>% mutate_at(c("mat_mean", "map_mean", "seasonality_t_mean", "seasonality_p_mean", "maxsrad_mean","minvapr_mean"),scale))

m_no_minvapr_mean <- lmer(log10.sporeVolume ~ Life_style + 
               mat_mean + 
               map_mean + 
               seasonality_t_mean + 
               seasonality_p_mean + 
               maxsrad_mean + 
               #minvapr_mean+
             (1|phylum/order)+(1|SporeType),  REML=TRUE,data=trial %>% filter(!grepl("Multi",SporeType)) %>% mutate_at(c("mat_mean", "map_mean", "seasonality_t_mean", "seasonality_p_mean", "maxsrad_mean","minvapr_mean"),scale))



modelos<-list(m_all,m_no_life_style,
              m_no_mat_mean,m_no_map_mean,
              m_no_seasonality_t_mean,m_no_seasonality_p_mean,
              m_no_maxsrad_mean,m_no_minvapr_mean)
names(modelos)<-c("m_all","m_no_life_style",
              "m_no_mat_mean","m_no_map_mean",
              "m_no_seasonality_t_mean","m_no_seasonality_p_mean",
              "m_no_maxsrad_mean","m_no_minvapr_mean")

resultados<-
do.call("rbind",
lapply(modelos, function(x){y<-r2_nakagawa(x)
data.frame(R2_random_fixed=y$R2_conditional,
           R2_fixed=y$R2_marginal,
           Log_likelihood=logLik(x),
           AIC=AIC(x))} )
);resultados$df<-AIC(m_all,m_no_life_style,
              m_no_mat_mean,m_no_map_mean,
              m_no_seasonality_t_mean,m_no_seasonality_p_mean,
              m_no_maxsrad_mean,m_no_minvapr_mean)$df

resultados$dAIC<-resultados$AIC-resultados$AIC[1]
resultados$Model<-rownames(resultados);rownames(resultados)<-NULL
resultados<-resultados[,c(7,5,1:4,6)]
#resultados
```


```{r}
dust(resultados, caption='Relative importance of lifestyle vs climatic variables in explaining interspecific spore size variation. The fit of a mixed effect model including lifestyle and six climatic variables as fixed factors (in italics) was compared relative to the fit of models in which one of these predictors was removed (indicated in the respective row). In all cases taxonomy and spore type were kept as random factors. df = Degrees of freedom , AIC = Akaike_s Information Criterion, dAIC= delta AIC (difference between the AIC of each and the one containing all terms). A large delta AIC indicates that dropping that term from the model results in a large decline in model fit. A small (< 2) or negative delta AIC indicates that dropping that term from the model improves model fit.') %>%
  sprinkle_print_method('markdown')  %>%
  sprinkle(rows=1, italic=TRUE) %>%
  sprinkle(cols=c(3,4), round=3) %>%
  sprinkle(cols=c(5,6,7), round=2)
```




## Spore aspect ratio
```{r, echo=FALSE}

### spore aspect ratio

## global model
m1 <- lm(log10.sporeAspectRatio ~ order + Life_style + SporeType + 
             mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial, na.action=na.fail)
# # plot to check for nonlinearities
# tmp <- m1$model %>% 
#   mutate_if(is.character, as.factor)
# for(i in 2:ncol(tmp)) plot(tmp$log10.sporeAspectRatio ~ tmp[, i], main=names(tmp)[i]) 

## results for individual variables
m1.var <- vector('list', length=ncol(m1$model))
names(m1.var) <- c('<none>', names(m1$model)[2:ncol(m1$model)])
m1.var[[1]] <- m1$model
for(i in 2:length(m1.var)) m1.var[[i]] <- m1$model[, -i]
m1.aic <- lapply(m1.var, function(x)AIC(m1, lm(log10.sporeAspectRatio ~ ., data=x)))
m1.drp <- data.frame(term = names(m1.var), 
                     Df = sapply(m1.aic, function(x)x[2, 'df']), 
                     AIC = sapply(m1.aic, function(x)x[2, 'AIC']), 
                     R2 = sapply(m1.var, function(x)summary(lm(log10.sporeAspectRatio ~ ., data=x))[['r.squared']])) %>% 
  mutate(type = c('global', rep('subset', n() - 1)), 
         dAIC = AIC - AIC[1], 
         dR2 = R2 - R2[1]) %>% 
  group_by(type) %>% 
  arrange(desc(dAIC), .by_group=TRUE) %>% 
  ungroup() %>% 
  select(term, Df, AIC, dAIC, R2, dR2)

## results using dredging
m1.drdg <- dredge(m1)
m1.drp <- left_join(m1.drp, data.frame(term = names(importance(m1.drdg)), 
                             sumWeights = importance(m1.drdg), row.names=NULL)) %>% 
  mutate(sumWeights = replace_na(sumWeights, '')) %>% 
  select(-R2, -dR2)

## results for variable groups
# drop taxonomy
m2.tax <- lm(log10.sporeAspectRatio ~ Life_style + SporeType + 
               mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop function
m2.fun <- lm(log10.sporeAspectRatio ~ order + SporeType + 
               mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop spore type
m2.typ <- lm(log10.sporeAspectRatio ~ order + Life_style + 
               mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop environment
m2.env <- lm(log10.sporeAspectRatio ~ order + Life_style + SporeType, data=trial, na.action=na.fail)
# make table
m2.drp <- data.frame(term=c('<none>', 'taxonomy', 'Life_style', 'SporeType', 'climate'), 
                     df=AIC(m1, m2.tax, m2.fun, m2.typ, m2.env)[['df']], 
                     AIC=AIC(m1, m2.tax, m2.fun, m2.typ, m2.env)[['AIC']], 
                     R2=sapply(list(m1, m2.tax, m2.fun, m2.typ, m2.env), function(x)summary(x)[['r.squared']]), 
                     row.names=NULL) %>% 
  mutate(type = c('<none>', rep('subset', n() - 1)), 
         dAIC = AIC - AIC[1], 
         dR2 = R2 - R2[1]) %>% 
  group_by(type) %>% 
  arrange(desc(dAIC), .by_group=TRUE) %>% 
  ungroup() %>% 
  select(term, df, AIC, dAIC, R2, dR2)

## results using dredging
m2.drdg <- list(m1, m2.tax, m2.fun, m2.typ, m2.env, 
                lm(log10.sporeAspectRatio ~ SporeType + mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial),  
                lm(log10.sporeAspectRatio ~ Life_style + mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial),  
                lm(log10.sporeAspectRatio ~ Life_style + SporeType, data=trial), 
                lm(log10.sporeAspectRatio ~ order + SporeType, data=trial),  
                lm(log10.sporeAspectRatio ~ order + mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial),  
                lm(log10.sporeAspectRatio ~ order + Life_style, data=trial),  
                lm(log10.sporeAspectRatio ~ SporeType, data=trial), 
                lm(log10.sporeAspectRatio ~ order, data=trial),  
                lm(log10.sporeAspectRatio ~ mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial),  
                lm(log10.sporeAspectRatio ~ Life_style, data=trial))
m2.drp <- left_join(m2.drp, data.frame(term = c('<none>', 'taxonomy', 'Life_style', 'SporeType', 'climate'), 
                             sumWeights = c(NA, importance(m2.drdg)[1:4]), row.names=NULL)) %>% 
  select(-R2, -dR2)

# Table 1
dust(m1.drp, caption='Table X. Importance of individual variables in linear models for explaining variation among species in spore shape Degrees of freedom (df), Akaike\'s Information Criterion (AIC) and delta AIC (dAIC) were calculated by dropping terms from a global model containing all terms (indicated in italics). Akaike weights (sumWeights) were calculated from AIC scores following automated model selection using all terms.') %>%
  sprinkle_print_method('markdown')  %>%
  sprinkle(rows=1, italic=TRUE) %>%
  sprinkle(cols=c(3, 4), round=1) %>%
  sprinkle(cols=5, round=2)

# Table 2
dust(m2.drp, caption='Table X. Importance of variable groupss in linear models for explaining variation among species in spore shape Degrees of freedom (df), Akaike\'s Information Criterion (AIC) and delta AIC (dAIC) were calculated by dropping groups of terms from a global model containing all terms (indicated in italics). Akaike weights (sumWeights) were calculated from AIC scores following automated model selection using all groups of terms.') %>%
  sprinkle_print_method('markdown')  %>%
  sprinkle(rows=1, italic=TRUE) %>%
  sprinkle(cols=c(3, 4), round=1) %>%
  sprinkle(cols=5, round=2)
  
```

Spore shape shows a similar pattern with regard to taxonomy being the most important, with spore type being second most importance. Climate and lifestyle ends up being of similar importance. The most important climate variable is solar radiation, with species with more ovoid spores being found in environments with higher solar radiation.


3. How much life-style influence the expected spore size ~ geographic extent?
This is another story :)


```{r}

temp<-left_join(read_csv('output/df_species_byPrimerSet.csv') %>%
                   mutate(names_to_use = gsub("_", " ", Species)) %>%
                   select(names_to_use, host.assoc, Primers.name, log10.sporeVolume, log10.sporeAspectRatio, starts_with('geo_'), ends_with('_sd'), n_samples_env),
                 To_Analysis[,c("names_to_use","SporeType","simpleFunct","Life_style",
                                "phylum","class","order")]) %>% 
  filter(!order == 'Not assigned') %>% 
  mutate(Life_style = recode(Life_style, Plant = "Facultative association"), 
         Life_style = ordered(Life_style), 
         Life_style = poly(as.numeric(Life_style, 1)))

temp<-left_join(read_csv('output/df_species_byPrimerSet.csv') %>%
                   mutate(names_to_use = gsub("_", " ", Species)) %>%
                   select(names_to_use, host.assoc, Primers.name, log10.sporeVolume, log10.sporeAspectRatio, starts_with('geo_'), ends_with('_sd'), n_samples_env),
                 To_Analysis[,c("names_to_use","SporeType","simpleFunct","Life_style",
                                "phylum","class","order")]) %>% 
  filter(!order == 'Not assigned') %>% 
  mutate(Life_style = recode(Life_style, Plant = "Facultative association"))

```


## Spore volume

```{r, eval=FALSE}

library(brms)

# m1 <- lmer(car::logit(geo_area_prop) ~ log10.sporeVolume * Life_style + 
#                  (1|order) + (1|Primers.name) + (1|SporeType), data=temp)
# m2 <- lmer(car::logit(geo_area_prop) ~ log10.sporeVolume * Life_style + 
#                  (1|order/Life_style) + (1|Primers.name) + (1|SporeType), data=temp)
# anova(m2, m1)

fit1.ext <- brm(formula = car::logit(geo_extent_prop) ~ log10.sporeVolume * Life_style + 
                  (1|order) + (1|Primers.name) + (1|SporeType),
                data=drop_na(select(temp, geo_extent_prop, log10.sporeVolume, Life_style, order, Primers.name, SporeType)),
                family = gaussian(),
                prior = c(set_prior("normal(0,5)", class = "b"),
                          set_prior("student_t(10,0,1)", class = "sd")),
                warmup = 1000, iter = 2000, chains = 4,
                control = list(adapt_delta = 0.995))
fit1.ext.loo <- loo(fit1.ext)
fit1.area <- brm(formula = car::logit(geo_area_prop) ~ log10.sporeVolume * Life_style + 
                  (1|order) + (1|Primers.name) + (1|SporeType),
                data=drop_na(select(temp, geo_area_prop, log10.sporeVolume, Life_style, order, Primers.name, SporeType)),
                family = gaussian(),
                prior = c(set_prior("normal(0,5)", class = "b"),
                          set_prior("student_t(10,0,1)", class = "sd")),
                warmup = 1000, iter = 2000, chains = 4,
                control = list(adapt_delta = 0.995))
fit1.area.loo <- loo(fit1.area)

# save(fit1.ext, fit1.ext.loo, fit1.area, fit1.area.loo, file='output/brmsoutput_sporeVolume.RData')

```


```{r}

library(ggridges)
library(patchwork)

load('output/brmsoutput_sporeVolume.RData')

### confidence bands for slope

# ext
slp.a <- c(attr(fit1.ext$fit, 'sim')$samples[[1]][['b_log10.sporeVolume']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[2]][['b_log10.sporeVolume']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[3]][['b_log10.sporeVolume']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[4]][['b_log10.sporeVolume']][1001:2000])
conf.95.a <- slp.a > quantile(slp.a, prob=0.025) & slp.a < quantile(slp.a, prob=0.975)
conf.50.a <- slp.a > quantile(slp.a, prob=0.25) & slp.a < quantile(slp.a, prob=0.75)
slp.b <- c(attr(fit1.ext$fit, 'sim')$samples[[1]][['b_log10.sporeVolume:Life_styleB_Opportunisticassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[2]][['b_log10.sporeVolume:Life_styleB_Opportunisticassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[3]][['b_log10.sporeVolume:Life_styleB_Opportunisticassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[4]][['b_log10.sporeVolume:Life_styleB_Opportunisticassociation']][1001:2000])
conf.95.b <- slp.b > quantile(slp.b, prob=0.025) & slp.b < quantile(slp.b, prob=0.975)
conf.50.b <- slp.b > quantile(slp.b, prob=0.25) & slp.b < quantile(slp.b, prob=0.75)
slp.c <- c(attr(fit1.ext$fit, 'sim')$samples[[1]][['b_log10.sporeVolume:Life_styleFacultativeassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[2]][['b_log10.sporeVolume:Life_styleFacultativeassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[3]][['b_log10.sporeVolume:Life_styleFacultativeassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[4]][['b_log10.sporeVolume:Life_styleFacultativeassociation']][1001:2000])
conf.95.c <- slp.c > quantile(slp.c, prob=0.025) & slp.c < quantile(slp.c, prob=0.975)
conf.50.c <- slp.c > quantile(slp.c, prob=0.25) & slp.c < quantile(slp.c, prob=0.75)
slp.d <- c(attr(fit1.ext$fit, 'sim')$samples[[1]][['b_log10.sporeVolume:Life_styleObligateassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[2]][['b_log10.sporeVolume:Life_styleObligateassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[3]][['b_log10.sporeVolume:Life_styleObligateassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[4]][['b_log10.sporeVolume:Life_styleObligateassociation']][1001:2000])
conf.95.d <- slp.d > quantile(slp.d, prob=0.025) & slp.d < quantile(slp.d, prob=0.975)
conf.50.d <- slp.d > quantile(slp.d, prob=0.25) & slp.d < quantile(slp.d, prob=0.75)
slp.ext <- bind_rows(data.frame(Life_style='free-living', response='extent', slope=slp.a, conf.50=conf.50.a, conf.95=conf.95.a), 
                   data.frame(Life_style='opportunistic', response='extent', slope=slp.a+slp.b, conf.50=conf.50.b, conf.95=conf.95.b), 
                   data.frame(Life_style='facultative', response='extent', slope=slp.a+slp.c, conf.50=conf.50.c, conf.95=conf.95.c), 
                   data.frame(Life_style='obligate', response='extent', slope=slp.a+slp.d, conf.50=conf.50.d, conf.95=conf.95.d))
rm(slp.a, slp.b, slp.c, slp.d, conf.95.a, conf.50.a, conf.95.b, conf.50.b, conf.95.c, conf.50.c, conf.95.d, conf.50.d)

# area
slp.a <- c(attr(fit1.area$fit, 'sim')$samples[[1]][['b_log10.sporeVolume']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[2]][['b_log10.sporeVolume']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[3]][['b_log10.sporeVolume']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[4]][['b_log10.sporeVolume']][1001:2000])
conf.95.a <- slp.a > quantile(slp.a, prob=0.025) & slp.a < quantile(slp.a, prob=0.975)
conf.50.a <- slp.a > quantile(slp.a, prob=0.25) & slp.a < quantile(slp.a, prob=0.75)
slp.b <- c(attr(fit1.area$fit, 'sim')$samples[[1]][['b_log10.sporeVolume:Life_styleB_Opportunisticassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[2]][['b_log10.sporeVolume:Life_styleB_Opportunisticassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[3]][['b_log10.sporeVolume:Life_styleB_Opportunisticassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[4]][['b_log10.sporeVolume:Life_styleB_Opportunisticassociation']][1001:2000])
conf.95.b <- slp.b > quantile(slp.b, prob=0.025) & slp.b < quantile(slp.b, prob=0.975)
conf.50.b <- slp.b > quantile(slp.b, prob=0.25) & slp.b < quantile(slp.b, prob=0.75)
slp.c <- c(attr(fit1.area$fit, 'sim')$samples[[1]][['b_log10.sporeVolume:Life_styleFacultativeassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[2]][['b_log10.sporeVolume:Life_styleFacultativeassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[3]][['b_log10.sporeVolume:Life_styleFacultativeassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[4]][['b_log10.sporeVolume:Life_styleFacultativeassociation']][1001:2000])
conf.95.c <- slp.c > quantile(slp.c, prob=0.025) & slp.c < quantile(slp.c, prob=0.975)
conf.50.c <- slp.c > quantile(slp.c, prob=0.25) & slp.c < quantile(slp.c, prob=0.75)
slp.d <- c(attr(fit1.area$fit, 'sim')$samples[[1]][['b_log10.sporeVolume:Life_styleObligateassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[2]][['b_log10.sporeVolume:Life_styleObligateassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[3]][['b_log10.sporeVolume:Life_styleObligateassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[4]][['b_log10.sporeVolume:Life_styleObligateassociation']][1001:2000])
conf.95.d <- slp.d > quantile(slp.d, prob=0.025) & slp.d < quantile(slp.d, prob=0.975)
conf.50.d <- slp.d > quantile(slp.d, prob=0.25) & slp.d < quantile(slp.d, prob=0.75)
slp.area <- bind_rows(data.frame(Life_style='free-living', response='area', slope=slp.a, conf.50=conf.50.a, conf.95=conf.95.a), 
                   data.frame(Life_style='opportunistic', response='area', slope=slp.a+slp.b, conf.50=conf.50.b, conf.95=conf.95.b), 
                   data.frame(Life_style='facultative', response='area', slope=slp.a+slp.c, conf.50=conf.50.c, conf.95=conf.95.c), 
                   data.frame(Life_style='obligate', response='area', slope=slp.a+slp.d, conf.50=conf.50.d, conf.95=conf.95.d))
rm(slp.a, slp.b, slp.c, slp.d, conf.95.a, conf.50.a, conf.95.b, conf.50.b, conf.95.c, conf.50.c, conf.95.d, conf.50.d)

# join
slp <- bind_rows(slp.ext, slp.area) %>% 
  mutate(Life_style = factor(Life_style, levels=c('free-living', 'opportunistic', 'facultative', 'obligate')))
slp %>% group_by(response, Life_style) %>% summarise(mean=mean(slope))


### plot
labs <- c(area = 'Range area', extent = 'Maximum distance')

# extent and area by volume
temp %>% 
  mutate(Life_style = recode(Life_style, `AFree living`='free-living', `B_Opportunistic association`='opportunistic', 
                             `Facultative association`='facultative', `Obligate association`='obligate'), 
         Life_style = factor(Life_style, levels=c('free-living', 'opportunistic', 'facultative', 'obligate'))) %>% 
  select(order, log10.sporeVolume, Life_style, geo_extent_prop, geo_area_prop) %>% 
  pivot_longer(cols=ends_with('prop'), names_to='response', values_to='value') %>% 
  ggplot(aes(x=log10.sporeVolume, y=car::logit(value), colour=Life_style)) + 
  geom_point(alpha=0.5, shape=16) + 
  stat_smooth(method='lm', alpha=0.25) + 
  labs(x='Spore volume (um^3, log_10)', y='Geographic extent (relative to maximum distance/area, logit)') + 
  facet_grid(rows=vars(response), scales='free_y') + 
  scale_colour_manual(name='', values=colorRampPalette(c('black', 'orange'))(4)) +
  theme(legend.position='none', 
        strip.background = element_blank(),
        strip.text.y = element_blank()) -> p1

ggplot(slp, aes(x=slope, colour=Life_style, fill=Life_style)) + 
  geom_density(alpha=0.5) + 
  geom_vline(xintercept=0) + 
  labs(x='Slope estimate', y='Density') + 
  facet_grid(rows=vars(response), labeller=labeller(response=labs)) + 
  scale_colour_manual(name='', values=colorRampPalette(c('black', 'orange'))(4)) + 
  scale_fill_manual(name='', values=colorRampPalette(c('black', 'orange'))(4)) -> p2

p1 + p2



```


## Spore aspect ratio

```{r, eval=FALSE}

library(brms)

fit1.ext <- brm(formula = car::logit(geo_extent_prop) ~ log10.sporeAspectRatio * Life_style + 
                  (1|order) + (1|Primers.name) + (1|SporeType),
                data=drop_na(select(temp, geo_extent_prop, log10.sporeAspectRatio, Life_style, order, Primers.name, SporeType)),
                family = gaussian(),
                prior = c(set_prior("normal(0,5)", class = "b"),
                          set_prior("student_t(10,0,1)", class = "sd")),
                warmup = 1000, iter = 2000, chains = 4,
                control = list(adapt_delta = 0.995))
fit1.ext.loo <- loo(fit1.ext)
fit1.area <- brm(formula = car::logit(geo_area_prop) ~ log10.sporeAspectRatio * Life_style + 
                  (1|order) + (1|Primers.name) + (1|SporeType),
                data=drop_na(select(temp, geo_area_prop, log10.sporeAspectRatio, Life_style, order, Primers.name, SporeType)),
                family = gaussian(),
                prior = c(set_prior("normal(0,5)", class = "b"),
                          set_prior("student_t(10,0,1)", class = "sd")),
                warmup = 1000, iter = 2000, chains = 4,
                control = list(adapt_delta = 0.995))
fit1.area.loo <- loo(fit1.area)

# save(fit1.ext, fit1.ext.loo, fit1.area, fit1.area.loo, file='output/brmsoutput_sporeAspectRatio.RData')

```


```{r}

library(ggridges)
library(patchwork)

load('output/brmsoutput_sporeAspectRatio.RData')

### confidence bands for slope

# ext
slp.a <- c(attr(fit1.ext$fit, 'sim')$samples[[1]][['b_log10.sporeAspectRatio']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[2]][['b_log10.sporeAspectRatio']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[3]][['b_log10.sporeAspectRatio']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[4]][['b_log10.sporeAspectRatio']][1001:2000])
conf.95.a <- slp.a > quantile(slp.a, prob=0.025) & slp.a < quantile(slp.a, prob=0.975)
conf.50.a <- slp.a > quantile(slp.a, prob=0.25) & slp.a < quantile(slp.a, prob=0.75)
slp.b <- c(attr(fit1.ext$fit, 'sim')$samples[[1]][['b_log10.sporeAspectRatio:Life_styleB_Opportunisticassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[2]][['b_log10.sporeAspectRatio:Life_styleB_Opportunisticassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[3]][['b_log10.sporeAspectRatio:Life_styleB_Opportunisticassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[4]][['b_log10.sporeAspectRatio:Life_styleB_Opportunisticassociation']][1001:2000])
conf.95.b <- slp.b > quantile(slp.b, prob=0.025) & slp.b < quantile(slp.b, prob=0.975)
conf.50.b <- slp.b > quantile(slp.b, prob=0.25) & slp.b < quantile(slp.b, prob=0.75)
slp.c <- c(attr(fit1.ext$fit, 'sim')$samples[[1]][['b_log10.sporeAspectRatio:Life_styleFacultativeassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[2]][['b_log10.sporeAspectRatio:Life_styleFacultativeassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[3]][['b_log10.sporeAspectRatio:Life_styleFacultativeassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[4]][['b_log10.sporeAspectRatio:Life_styleFacultativeassociation']][1001:2000])
conf.95.c <- slp.c > quantile(slp.c, prob=0.025) & slp.c < quantile(slp.c, prob=0.975)
conf.50.c <- slp.c > quantile(slp.c, prob=0.25) & slp.c < quantile(slp.c, prob=0.75)
slp.d <- c(attr(fit1.ext$fit, 'sim')$samples[[1]][['b_log10.sporeAspectRatio:Life_styleObligateassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[2]][['b_log10.sporeAspectRatio:Life_styleObligateassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[3]][['b_log10.sporeAspectRatio:Life_styleObligateassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[4]][['b_log10.sporeAspectRatio:Life_styleObligateassociation']][1001:2000])
conf.95.d <- slp.d > quantile(slp.d, prob=0.025) & slp.d < quantile(slp.d, prob=0.975)
conf.50.d <- slp.d > quantile(slp.d, prob=0.25) & slp.d < quantile(slp.d, prob=0.75)
slp.ext <- bind_rows(data.frame(Life_style='free-living', response='extent', slope=slp.a, conf.50=conf.50.a, conf.95=conf.95.a), 
                   data.frame(Life_style='opportunistic', response='extent', slope=slp.a+slp.b, conf.50=conf.50.b, conf.95=conf.95.b), 
                   data.frame(Life_style='facultative', response='extent', slope=slp.a+slp.c, conf.50=conf.50.c, conf.95=conf.95.c), 
                   data.frame(Life_style='obligate', response='extent', slope=slp.a+slp.d, conf.50=conf.50.d, conf.95=conf.95.d))
rm(slp.a, slp.b, slp.c, slp.d, conf.95.a, conf.50.a, conf.95.b, conf.50.b, conf.95.c, conf.50.c, conf.95.d, conf.50.d)

# area
slp.a <- c(attr(fit1.area$fit, 'sim')$samples[[1]][['b_log10.sporeAspectRatio']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[2]][['b_log10.sporeAspectRatio']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[3]][['b_log10.sporeAspectRatio']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[4]][['b_log10.sporeAspectRatio']][1001:2000])
conf.95.a <- slp.a > quantile(slp.a, prob=0.025) & slp.a < quantile(slp.a, prob=0.975)
conf.50.a <- slp.a > quantile(slp.a, prob=0.25) & slp.a < quantile(slp.a, prob=0.75)
slp.b <- c(attr(fit1.area$fit, 'sim')$samples[[1]][['b_log10.sporeAspectRatio:Life_styleB_Opportunisticassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[2]][['b_log10.sporeAspectRatio:Life_styleB_Opportunisticassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[3]][['b_log10.sporeAspectRatio:Life_styleB_Opportunisticassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[4]][['b_log10.sporeAspectRatio:Life_styleB_Opportunisticassociation']][1001:2000])
conf.95.b <- slp.b > quantile(slp.b, prob=0.025) & slp.b < quantile(slp.b, prob=0.975)
conf.50.b <- slp.b > quantile(slp.b, prob=0.25) & slp.b < quantile(slp.b, prob=0.75)
slp.c <- c(attr(fit1.area$fit, 'sim')$samples[[1]][['b_log10.sporeAspectRatio:Life_styleFacultativeassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[2]][['b_log10.sporeAspectRatio:Life_styleFacultativeassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[3]][['b_log10.sporeAspectRatio:Life_styleFacultativeassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[4]][['b_log10.sporeAspectRatio:Life_styleFacultativeassociation']][1001:2000])
conf.95.c <- slp.c > quantile(slp.c, prob=0.025) & slp.c < quantile(slp.c, prob=0.975)
conf.50.c <- slp.c > quantile(slp.c, prob=0.25) & slp.c < quantile(slp.c, prob=0.75)
slp.d <- c(attr(fit1.area$fit, 'sim')$samples[[1]][['b_log10.sporeAspectRatio:Life_styleObligateassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[2]][['b_log10.sporeAspectRatio:Life_styleObligateassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[3]][['b_log10.sporeAspectRatio:Life_styleObligateassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[4]][['b_log10.sporeAspectRatio:Life_styleObligateassociation']][1001:2000])
conf.95.d <- slp.d > quantile(slp.d, prob=0.025) & slp.d < quantile(slp.d, prob=0.975)
conf.50.d <- slp.d > quantile(slp.d, prob=0.25) & slp.d < quantile(slp.d, prob=0.75)
slp.area <- bind_rows(data.frame(Life_style='free-living', response='area', slope=slp.a, conf.50=conf.50.a, conf.95=conf.95.a), 
                   data.frame(Life_style='opportunistic', response='area', slope=slp.a+slp.b, conf.50=conf.50.b, conf.95=conf.95.b), 
                   data.frame(Life_style='facultative', response='area', slope=slp.a+slp.c, conf.50=conf.50.c, conf.95=conf.95.c), 
                   data.frame(Life_style='obligate', response='area', slope=slp.a+slp.d, conf.50=conf.50.d, conf.95=conf.95.d))
rm(slp.a, slp.b, slp.c, slp.d, conf.95.a, conf.50.a, conf.95.b, conf.50.b, conf.95.c, conf.50.c, conf.95.d, conf.50.d)

# join
slp <- bind_rows(slp.ext, slp.area) %>% 
  mutate(Life_style = factor(Life_style, levels=c('free-living', 'opportunistic', 'facultative', 'obligate')))
slp %>% group_by(response, Life_style) %>% summarise(mean=mean(slope))


### plot
labs <- c(area = 'Range area', extent = 'Maximum distance')

# extent and area by volume
temp %>% 
  mutate(Life_style = recode(Life_style, `AFree living`='free-living', `B_Opportunistic association`='opportunistic', 
                             `Facultative association`='facultative', `Obligate association`='obligate'), 
         Life_style = factor(Life_style, levels=c('free-living', 'opportunistic', 'facultative', 'obligate'))) %>% 
  select(order, log10.sporeAspectRatio, Life_style, geo_extent_prop, geo_area_prop) %>% 
  pivot_longer(cols=ends_with('prop'), names_to='response', values_to='value') %>% 
  ggplot(aes(x=log10.sporeAspectRatio, y=car::logit(value), colour=Life_style)) + 
  geom_point(alpha=0.5, shape=16) + 
  stat_smooth(method='lm', alpha=0.25) + 
  labs(x='Spore aspect ratio (length / width, log_10)', y='Geographic extent (relative to maximum distance/area, logit)') + 
  facet_grid(rows=vars(response), scales='free_y') + 
  scale_colour_manual(name='', values=colorRampPalette(c('black', 'orange'))(4)) +
  theme(legend.position='none', 
        strip.background = element_blank(),
        strip.text.y = element_blank()) -> p1

ggplot(slp, aes(x=slope, colour=Life_style, fill=Life_style)) + 
  geom_density(alpha=0.5) + 
  geom_vline(xintercept=0) + 
  labs(x='Slope estimate', y='Density') + 
  facet_grid(rows=vars(response), labeller=labeller(response=labs)) + 
  scale_colour_manual(name='', values=colorRampPalette(c('black', 'orange'))(4)) + 
  scale_fill_manual(name='', values=colorRampPalette(c('black', 'orange'))(4)) -> p2

p1 + p2



```