---
title: "SummarySporeTraitAnalysis"
author: "Carlos"
date: "22/04/2020"
output:
  html_document: default
  word_document: default
---


```{r, message=FALSE, warning=FALSE}
#library(smatr)
library(performance)
library(tidyverse)
library(lme4)
library(lmerTest)
library(MuMIn)
library(broom)
library(pixiedust)
source("MatchingSpore_FunctionData.R")
```



```{r, echo=FALSE,warning=FALSE}
library(tidyverse)
library(phylolm)


#Genome wide phylogenetic tree
Li_tree<-read.tree("Li_etal_tree.treefile")

#Spore and functional data for the species present in the tree
Li_tree_names<-readRDS("Li_tree_names.RDS")
#Excluding Chytrids and Entomophthoromycota (because there´s only 1 record for each)
Li_tree_names<-Li_tree_names[-which(Li_tree_names$phylum=="Chytridiomycota"|
                                      Li_tree_names$phylum=="Entomophthoromycota"),]





genera<-unique(Li_tree_names$genus)

#All species from which we have spore and functional data

To_Analysis$simpleFunct<-gsub("Plant-Human","Human",To_Analysis$simpleFunct)
To_Analysis$Life_style<-gsub("Plant-Human","B_Opportunistic association",To_Analysis$Life_style)

To_Analysis<-To_Analysis[-grep("-",To_Analysis$Life_style),]

To_Analysis$simpleFunct<-gsub("Saprotroph","A_Saprotroph",To_Analysis$simpleFunct)
To_Analysis$Life_style<-gsub("Plant","Facultative association",To_Analysis$Life_style)

#unique(To_Analysis$simpleFunct)
#unique(To_Analysis$Life_style)

#All species from which we have spore, functional, geographic extent and climate data
dat <- read_csv('output/df_species_noPrimers.csv')



```

Cladograms based on taxonomy
```{r}


frm <- ~phylum/class/order/family/genus/names_to_use

#Just using the data with both functional and spore data
To_Analysis1<-To_Analysis
To_Analysis1<-To_Analysis1[-which(To_Analysis1$simpleFunct=="Animal"),]
To_Analysis1<-To_Analysis1[-which(To_Analysis1$simpleFunct=="Fungi"),]
To_Analysis1<-To_Analysis1[-which(duplicated(To_Analysis1$names_to_use)),]

#Using taxonomic trees would require different trees for each analysis:
#1. For all the data in To_analysis to see whether spore size correlates with symbiotic status

tree_To_Analysis <- as.phylo(frm, data = To_Analysis1 %>%
                 mutate_at(c("phylum","class", "order",
                             "family","genus","names_to_use"),as.factor))


is.binary.tree(tree_To_Analysis)

tree_To_Analysis<-multi2di(tree_To_Analysis)
is.binary.tree(tree_To_Analysis)

tree_To_Analysis$edge.length<-rep(1, nrow(tree_To_Analysis$edge))
tax_tree<-tree_To_Analysis;rm(tree_To_Analysis,To_Analysis1)


#Using the species from which we have spore data (many more than above)
tax_tree2 <- as.phylo(frm, data = Spore_data[-which(duplicated(Spore_data$names_to_use)),] %>%
                 mutate_at(c("phylum","class", "order",
                             "family","genus","names_to_use"),as.factor))


tax_tree2<-multi2di(tax_tree2)
is.binary.tree(tax_tree2)

#Here what I am doing is giving the same length to all edges
tax_tree2$edge.length<-rep(1, nrow(tax_tree2$edge))
```



# About the datasets

There are 4 datasets with fungal data:

1. AllFungi: contains all info on the names and taxonomy used (the source of taxonomy, which is mostly CoL), the descriptions id, the spore types and the spore names (conidia, chlamydospores, basidiospores, etc). This dataset is basically the result of the assembly of all spore data.

2. Spore_data: It contains simplified information on spore dimensions for all fungi. That is, I summarized entries repeated because different descriptions ID and/or species having different names. I also summarized the sporeName to only considers 4 spore type categories as we discuss in the manuscript (meiosproes, mitospores, multinucleate sexual and multinucleate asexual spores).

3. "Spore_functions"  contains all the info as in Spore_data plus functional data (if any). For species without functional data, the entries have NAs

4. To_Analysis: It contians information for the subset of species for which we have both spore data (spore data from Spore_data) and functional data


# Statistical analysis



Counting how much information we have
```{r}
#unique species names collected through data assembly
length(unique(AllFungi$base_name))

#unique species names obtained only through text mining from Mycobank
length(unique(AllFungi$base_name[which(AllFungi$source_base_name=="Mycobank")]))

#unique species names complemented by other sources
length(
  unique(AllFungi$base_name[which(AllFungi$source_base_name!="Mycobank")])[-which(
    unique(AllFungi$base_name[which(AllFungi$source_base_name!="Mycobank")])%in%unique(AllFungi$base_name[which(AllFungi$source_base_name=="Mycobank")])
  )]
  
  
)

#unique accepted names after correcting for taxonomy in the Catalogue of Life (accessed on December 2019)
length(unique(AllFungi$names_to_use))


#unique accepted names for which we have functional trait data
length(unique(To_Analysis$names_to_use))

length(unique(To_Analysis$names_to_use))/length(unique(AllFungi$names_to_use))

```



# Differences among spore types

AM spores vs other asexual spores
```{r}
library(lme4)
library(performance)

#Li dataset

dat2<-Li_tree_names[which(Li_tree_names$SporeType=="Mitospores"|
                         Li_tree_names$SporeType=="Multinucleate asexual spores"),]

rownames(dat2)<-dat2$orginal_tree_names

AM_vs_Mito_Li<-
  phylolm(log10(SporeVolume) ~ SporeType,
          data = dat2,
         
          phy = Li_tree, model = "lambda")

rm(dat2)

#Cladogram dataset

dat1<-Spore_data[which(Spore_data$SporeType=="Mitospores"|
                         Spore_data$SporeType=="Multinucleate asexual spores"),]

dat1<-dat1[-grep("Zygomy",dat1$phylum),]
rownames(dat1)<-dat1$names_to_use

AM_vs_Mito<-
phylolm(log10(SporeVolume)~SporeType,
        phy = tax_tree2,
        model = "lambda",
        data = dat1)

Simple_AM_vs_Mito<-
lm(log10(SporeVolume)~SporeType,
        #phy = tree_Spore_data,
        #model = "lambda",
        data = dat1)

```

Comparisons made with Li subset
```{r}
summary(AM_vs_Mito_Li)
```

Comparison made with cladograms
```{r}
summary(AM_vs_Mito)
```

No correction at all
```{r}
summary(Simple_AM_vs_Mito)
```

In this case AM Spores are 3.5 order of magnitude larger than mitospores. A result that is consistent in all models


AM spores vs sexual spores

```{r}
#AM_vs_Meiospores
dat2<-Li_tree_names[which(Li_tree_names$SporeType=="Meiospores"|
                         Li_tree_names$SporeType=="Multinucleate asexual spores"),]

rownames(dat2)<-dat2$orginal_tree_names

AM_vs_Meio_Li<-
  phylolm(log10(SporeVolume) ~ SporeType,
          data = dat2,
         
          phy = Li_tree, model = "lambda")

rm(dat2)

#cladogram
dat1<-Spore_data[which(Spore_data$SporeType=="Meiospores"|
                         Spore_data$SporeType=="Multinucleate asexual spores"),]

dat1<-dat1[-grep("Zygomy",dat1$phylum),]
rownames(dat1)<-dat1$names_to_use

# pruned.tree_1<-drop.tip(tree_Spore_data,
#                 tree_Spore_data$tip.label[!tree_Spore_data$tip.label%in%dat1$names_to_use])

AM_vs_Meio<-
phylolm(log10(SporeVolume)~SporeType,
        phy = tax_tree2,
        model = "lambda",
        data = dat1)


Simple_AM_vs_Meio<-
lm(log10(SporeVolume)~SporeType,
        #phy = tree_Spore_data,
        #model = "lambda",
        data = dat1)
#summary(Simple_AM_vs_Meio)
```

Comparison using the Li subset
```{r}
summary(AM_vs_Meio_Li)
```

Comparison using the cladogram
```{r}
summary(AM_vs_Meio)
```

Not incorporating phylogeny
```{r}
summary(Simple_AM_vs_Meio)
```


Comparisons with Zygos (less important to have super nice)

```{r}
##AM_vs_Multinucleate sexual (zygo ones)

dat2<-Li_tree_names[grep("Zygomy|Glomero",Li_tree_names$phylum),]

dat2<-dat2[-which(grepl("Zygomy",dat2$phylum)&
                         dat2$SporeType=="Mitospores"),]

dat2$SporeType[which(dat2$SporeType=="Multinucleate sexual spores")]<-"A_Zygo spores"
dat2$SporeType[which(dat2$SporeType=="Multinucleate asexual spores")]<-"B_AM spores"

rownames(dat2)<-dat2$orginal_tree_names

#####

dat1<-Spore_data[grep("Zygomy|Glomero",Spore_data$phylum),]
dat1<-dat1[-which(dat1$SporeType=="Mitospores"),]

dat1$SporeType[grep("Zygomy",dat1$phylum)]<-"A_Zygo spores"
dat1$SporeType[grep("Glomero",dat1$phylum)]<-"B_AM spores"

dat1<-dat1[-which(duplicated(dat1$names_to_use)),]

rownames(dat1)<-dat1$names_to_use


# pruned.tree_1<-drop.tip(tree_Spore_data,
#                 tree_Spore_data$tip.label[!tree_Spore_data$tip.label%in%dat1$names_to_use])

AM_vs_ZygoS_Li<-
phylolm(log10(SporeVolume)~SporeType,
        phy = Li_tree,
        model = "lambda",
        data = dat2)

AM_vs_ZygoS<-
phylolm(log10(SporeVolume)~SporeType,
        phy = tax_tree2,
        model = "lambda",
        data = dat1)

Simple_AM_vs_ZygoS<-
lm(log10(SporeVolume)~SporeType,
        #phy = tree_Spore_data,
        #model = "lambda",
        data = dat1)

```


```{r}
summary(AM_vs_ZygoS_Li)
```

with the cladograms
```{r}
summary(AM_vs_ZygoS)
```

without phylogenetic corrections
```{r}
summary(Simple_AM_vs_ZygoS)
```



Basidiospores vs Ascospores

```{r}
meiospores<-Li_tree_names[which(Li_tree_names$SporeType=="Meiospores"),]

meiospores$Type<-NA
meiospores$Type[grep("Asco",meiospores$phylum)]<-"ascospores"
meiospores$Type[grep("Basidio",meiospores$phylum)]<-"basidiospores"

rownames(meiospores)<-meiospores$orginal_tree_names

asco_vs_basidio_Li<-
phylolm(log10(SporeVolume)~Type,
        phy = Li_tree,
        model = "lambda",
        data = meiospores);rm(meiospores)



dat1<-Spore_data[which(Spore_data$SporeType=="Meiospores"),]

dat1$Type<-NA
dat1$Type[grep("Asco",dat1$phylum)]<-"ascospores"
dat1$Type[grep("Basidio",dat1$phylum)]<-"basidiospores"

rownames(dat1)<-dat1$names_to_use

asco_vs_basidio<-
phylolm(log10(SporeVolume)~Type,
        phy = tax_tree2,
        model = "lambda",
        data = dat1)


asco_vs_basidio_simple<-
lm(log10(SporeVolume)~Type,
        #phy = tree_Spore_data,
        #model = "lambda",
        data = dat1);rm(dat1)
```

Comparison made with the Li subset
```{r}
summary(asco_vs_basidio_Li)
```

Comparison made with the cladograms
```{r}
summary(asco_vs_basidio)
```

Comparisons without incorporating phylogeny
```{r}
summary(asco_vs_basidio_simple)
```


## Question 1: Does spore size co-varies with functional groups?

For asking this question I have to separate it into distinct spore types. Ideally, only sexual (meiospore) and asexual (mitospores). 

```{r, echo=FALSE}

mi<-which(Li_tree_names$SporeType=="Meiospores")
mo<-which(Li_tree_names$SporeType=="Mitospores")
mn<-which(Li_tree_names$SporeType=="Multinucleate sexual spores")
ma<-which(Li_tree_names$SporeType=="Multinucleate asexual spores")

#Meiospores
meiospores<-Li_tree_names[mi,]

rownames(meiospores)<-meiospores$orginal_tree_names

# pruned.tree_me<-drop.tip(Li_tree,
#                  Li_tree$tip.label[!Li_tree$tip.label%in%meiospores$orginal_tree_names])

#Mitospores
mitospores<-Li_tree_names[mo,]
rownames(mitospores)<-mitospores$orginal_tree_names

# pruned.tree_mo<-drop.tip(Li_tree,
#                        Li_tree$tip.label[!Li_tree$tip.label%in%mitospores$orginal_tree_names])

#Meiospores
meiospores2<-To_Analysis[grep("eiospores",To_Analysis$SporeType),]
rownames(meiospores2)<-meiospores2$names_to_use

# pruned.tree_me<-drop.tip(tree_To_Analysis,
#                 tree_To_Analysis$tip.label[!tree_To_Analysis$tip.label%in%meiospores$names_to_use])


#Mitospores
mitospores2<-To_Analysis[grep("itospores",To_Analysis$SporeType),]
rownames(mitospores2)<-mitospores2$names_to_use

# pruned.tree_mo<-drop.tip(tree_To_Analysis,
#                 tree_To_Analysis$tip.label[!tree_To_Analysis$tip.label%in%mitospores$names_to_use])

mod_Li_meios<-
  phylolm(log10(SporeVolume) ~ simpleFunct,
          data = meiospores,
         
          phy = Li_tree,
          #phy = Li_tree2,
          model = "lambda")

mod_tax_meios<-
  phylolm(log10(SporeVolume) ~ simpleFunct,
          data = meiospores2,
         
          phy = tax_tree,#tree_To_Analysis,#pruned.tree_me,#tax_tree,# #
          model = "lambda")


mod_Li_mitos<-phylolm(log10(SporeVolume) ~ simpleFunct,
              data = mitospores,
              phy = Li_tree, 
              #phy = Li_tree2,
              model = "lambda")


mod_tax_mitos<-phylolm(log10(SporeVolume) ~ simpleFunct,
              data = mitospores2,
              phy = tax_tree, model = "lambda")

```

For Meiospores

```{r}

summary(mod_Li_meios)
```


```{r}
summary(mod_tax_meios)
```

For Mitospores

```{r}
summary(mod_Li_mitos)
```


```{r}
summary(mod_tax_mitos)
```

## Question 2: Do more symbiotic fungi have larger spores?

I do this with a priori contrasts:

* Life_style1: free living against symbionts
* Life_style2: free living  vs opportunistic symbionts
* Life_style2: Faculative vs Obligate

```{r,echo=FALSE}
meiospores$Life_style<-as.factor(meiospores$Life_style)
#levels(meiospores$Life_style)
contrasts(meiospores$Life_style)<-cbind(
  c(-1,-1,1,1),#free living against symbionts
  c(-1,1,0,0),#free living  vs opportunistic symbionts
  c(0,0,-1,1)#Faculative vs Obligate
)

mitospores$Life_style<-as.factor(mitospores$Life_style)
#levels(mitospores$Life_style)
contrasts(mitospores$Life_style)<-cbind(
  c(-1,-1,1,1),#free living against symbionts
  c(-1,1,0,0),#free living  vs opportunistic symbionts
  c(0,0,-1,1)#Faculative vs Obligate
)

#################################################################
meiospores2<-meiospores2[-which(meiospores2$Life_style=="Fungi"),]
meiospores2$Life_style<-as.factor(meiospores2$Life_style)
levels(meiospores2$Life_style)
contrasts(meiospores2$Life_style)<-cbind(
  c(-1,-1,1,1),#free living against symbionts
  c(-1,1,0,0),#free living  vs opportunistic symbionts
  c(0,0,-1,1)#Faculative vs Obligate
)


mitospores2<-mitospores2[-which(mitospores2$Life_style=="Fungi"),]
mitospores2<-mitospores2[-which(mitospores2$Life_style=="Animal"),]

mitospores2$Life_style<-as.factor(mitospores2$Life_style)
levels(mitospores2$Life_style)
contrasts(mitospores2$Life_style)<-cbind(
  c(-1,-1,1,1),#free living against symbionts
  c(-1,1,0,0),#free living  vs opportunistic symbionts
  c(0,0,-1,1)#Faculative vs Obligate
)

mod_Li_meios_symb<-
  phylolm(log10(SporeVolume) ~ Life_style,
          data = meiospores,
          phy = Li_tree,
          #phy = Li_tree2,
          model = "lambda")

mod_Li_mitos_symb<-
  phylolm(log10(SporeVolume) ~ Life_style,
          data = mitospores,
          phy = Li_tree,
          #phy = Li_tree2,
          model = "lambda")

mod_tax_meios_symb<-
  phylolm(log10(SporeVolume) ~ Life_style,
          data = meiospores2,
          
          phy = tax_tree,
          #phy = Li_tree2,
          model = "lambda")

mod_tax_mitos_symb<-
  phylolm(log10(SporeVolume) ~ Life_style,
          data = mitospores2,
          
          phy = tax_tree2,
          #phy = Li_tree2,
          model = "lambda")

```

#### Are meiospores and mitospores bigger the more symbiotic they are?

first with the with the Li data
```{r}
summary(mod_Li_meios_symb)
```

With the 
```{r}
summary(mod_Li_mitos_symb)
```


```{r}
summary(mod_tax_meios_symb)
```


```{r}
summary(mod_tax_mitos_symb)
```




# 3.Determinig the relative importance of climate vs life style in explaining interpecific spore variation 

First with the Li tree

```{r}
dat$orginal_tree_names<-dat$Species

trial1<-left_join(
  Li_tree_names[,c("orginal_tree_names","SporeType","SporeVolume","simpleFunct","Life_style","phylum","class","order","family","genus")],
  dat %>%
    select(orginal_tree_names, host.assoc, log10.sporeVolume, log10.sporeAspectRatio, ends_with('mean'), n_samples_env))

trial1<-trial1[complete.cases(trial1),]

# this is weird that there are species without assignments
trial1<-trial1 %>% 
  #filter(!order == 'Not assigned') %>% 
  mutate(Life_style = recode(Life_style, Plant = "Facultative association"), 
         Life_style = recode(Life_style, "B_Opportunistic association" = "AFree living"), #Just added this
         Life_style = ordered(Life_style))#, 
         # Life_style = poly(as.numeric(Life_style, 1)))

trial1<-
trial1 %>%#filter(!grepl("Multi",SporeType)) %>%
                 mutate_at(c("mat_mean", "map_mean", "seasonality_t_mean", 
                             "seasonality_p_mean", "maxsrad_mean","minvapr_mean"),scale)
rownames(trial1)<-trial1$orginal_tree_names

# pruned.tree_trial1<-drop.tip(Li_tree,
#                 Li_tree$tip.label[!Li_tree$tip.label%in%trial1$orginal_tree_names])


m_all <- phylolm(log10(SporeVolume) ~ Life_style + mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean+ SporeType,
               phy = Li_tree,
               model = "lambda",
               data=trial1)
  
m_no_life_style <- phylolm(log10(SporeVolume) ~ #Life_style + 
               mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean+ SporeType,
               phy = Li_tree,
               model = "lambda",
               data=trial1)
  
m_no_mat_mean <- phylolm(log10(SporeVolume) ~ Life_style + 
               #mat_mean + 
                 map_mean + 
                 seasonality_t_mean + 
                 seasonality_p_mean + 
                 maxsrad_mean + 
                 minvapr_mean+
               SporeType,
               phy = Li_tree,
               model = "lambda",
               data=trial1)
  
  
  m_no_map_mean <- phylolm(log10(SporeVolume) ~ Life_style + 
                 mat_mean + 
                 #map_mean + 
                 seasonality_t_mean + 
                 seasonality_p_mean + 
                 maxsrad_mean + 
                 minvapr_mean+
               SporeType,
               phy = Li_tree,
               model = "lambda",
               data=trial1)
  
  m_no_seasonality_t_mean <- phylolm(log10(SporeVolume) ~ Life_style + 
                 mat_mean + 
                 map_mean + 
                 #seasonality_t_mean + 
                 seasonality_p_mean + 
                 maxsrad_mean + 
                 minvapr_mean+
                   SporeType,
               phy = Li_tree,
               model = "lambda",
               data=trial1)
  
  m_no_seasonality_p_mean <- phylolm(log10(SporeVolume) ~ Life_style + 
                 mat_mean + 
                 map_mean + 
                 seasonality_t_mean + 
                 #seasonality_p_mean + 
                 maxsrad_mean + 
                 minvapr_mean+
               SporeType,  
               phy = Li_tree,
               model = "lambda",
               data=trial1)
  
  m_no_maxsrad_mean <- phylolm(log10(SporeVolume) ~ Life_style + 
                 mat_mean + 
                 map_mean + 
                 seasonality_t_mean + 
                 seasonality_p_mean + 
                 #maxsrad_mean + 
                 minvapr_mean+
               SporeType,
               phy = Li_tree,
               model = "lambda",
               data=trial1)
  
  m_no_minvapr_mean <- phylolm(log10(SporeVolume) ~ Life_style + 
                 mat_mean + 
                 map_mean + 
                 seasonality_t_mean + 
                 seasonality_p_mean + 
                 maxsrad_mean + 
                 #minvapr_mean+
               SporeType,
               phy = Li_tree,
               model = "lambda",
               data=trial1)
  
  
  
  modelos<-list(m_all,m_no_life_style,
                m_no_mat_mean,m_no_map_mean,
                m_no_seasonality_t_mean,m_no_seasonality_p_mean,
                m_no_maxsrad_mean,m_no_minvapr_mean)
  names(modelos)<-c("m_all","m_no_life_style",
                "m_no_mat_mean","m_no_map_mean",
                "m_no_seasonality_t_mean","m_no_seasonality_p_mean",
                "m_no_maxsrad_mean","m_no_minvapr_mean")
  
  resultados<-
  do.call("rbind",
  lapply(modelos, function(x){#y<-r2_nakagawa(x)
  data.frame(adj.r.squared=summary(x)$adj.r.squared,
             #R2_fixed=y$R2_marginal,
    df=logLik.phylolm(x)[2],         
    Log_likelihood=logLik.phylolm(x)[1],
    AIC=AIC.phylolm(x))} ))
  
  resultados$dAIC<-resultados$AIC-resultados$AIC[1]
  resultados$Model<-rownames(resultados);rownames(resultados)<-NULL
  resultados<-resultados[,c(6,2,1,3:5)]
  resultados_Li_tree<-resultados
  
```



```{r, echo=FALSE,message=FALSE,warning=FALSE}
dat$names_to_use<-gsub("_"," ",dat$Species)

trial<-left_join(dat %>% 
                  select(names_to_use, host.assoc, log10.sporeVolume, log10.sporeAspectRatio, ends_with('mean'), n_samples_env), 
                 To_Analysis[,c("names_to_use","SporeType","SporeVolume","simpleFunct","Life_style",
                                "phylum","class","order","family","genus")])

trial<-trial[complete.cases(trial),]
nta<-
which(rowSums(sapply(trial[,c("phylum","class", "order","family")],function(x){x=="Not assigned"}))==1)

#trial<-trial[-nta,]
trial<-trial[complete.cases(trial),]

# this is weird that there are species without assignments
trial<-trial %>% 
  #filter(!order == 'Not assigned') %>% 
  mutate(Life_style = recode(Life_style, Plant = "Facultative association"), 
         Life_style = recode(Life_style, "B_Opportunistic association" = "AFree living"),
         Life_style = ordered(Life_style))

trial<-
trial %>%#filter(!grepl("Multi",SporeType)) %>%
                 mutate_at(c("mat_mean", "map_mean", "seasonality_t_mean", 
                             "seasonality_p_mean", "maxsrad_mean","minvapr_mean"),scale)

rownames(trial)<-trial$names_to_use

#2. One for the geographic data, that is for dat
frm <- ~phylum/class/order/family/genus/names_to_use

tree_trial <- as.phylo(frm, data = trial %>%
                 mutate_at(c("phylum","class", "order",
                             "family","genus","names_to_use"),as.factor))
is.binary.tree(tree_trial)

tree_trial<-multi2di(tree_trial)
is.binary.tree(tree_trial)

tree_trial$edge.length<-rep(1, nrow(tree_trial$edge))

#plot(tree_trial,cex = 0.01)#Now this work

m_all <- phylolm(log10(SporeVolume) ~ Life_style + mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean+ SporeType,
               phy = tree_trial,
               model = "lambda",
               data=trial)
  
m_no_life_style <- phylolm(log10(SporeVolume) ~ #Life_style + 
               mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean+ SporeType,
               phy = tree_trial,
               model = "lambda",
               data=trial)
  
m_no_mat_mean <- phylolm(log10(SporeVolume) ~ Life_style + 
               #mat_mean + 
                 map_mean + 
                 seasonality_t_mean + 
                 seasonality_p_mean + 
                 maxsrad_mean + 
                 minvapr_mean+
               SporeType,
               phy = tree_trial,
               model = "lambda",
               data=trial)
  
  
  m_no_map_mean <- phylolm(log10(SporeVolume) ~ Life_style + 
                 mat_mean + 
                 #map_mean + 
                 seasonality_t_mean + 
                 seasonality_p_mean + 
                 maxsrad_mean + 
                 minvapr_mean+
               SporeType,
               phy = tree_trial,
               model = "lambda",
               data=trial)
  
  m_no_seasonality_t_mean <- phylolm(log10(SporeVolume) ~ Life_style + 
                 mat_mean + 
                 map_mean + 
                 #seasonality_t_mean + 
                 seasonality_p_mean + 
                 maxsrad_mean + 
                 minvapr_mean+
                   SporeType,
               phy = tree_trial,
               model = "lambda",
               data=trial)
  
  m_no_seasonality_p_mean <- phylolm(log10(SporeVolume) ~ Life_style + 
                 mat_mean + 
                 map_mean + 
                 seasonality_t_mean + 
                 #seasonality_p_mean + 
                 maxsrad_mean + 
                 minvapr_mean+
               SporeType,  
               phy = tree_trial,
               model = "lambda",
               data=trial)
  
  m_no_maxsrad_mean <- phylolm(log10(SporeVolume) ~ Life_style + 
                 mat_mean + 
                 map_mean + 
                 seasonality_t_mean + 
                 seasonality_p_mean + 
                 #maxsrad_mean + 
                 minvapr_mean+
               SporeType,
              phy = tree_trial,
               model = "lambda",
               data=trial)
  
  m_no_minvapr_mean <- phylolm(log10(SporeVolume) ~ Life_style + 
                 mat_mean + 
                 map_mean + 
                 seasonality_t_mean + 
                 seasonality_p_mean + 
                 maxsrad_mean + 
                 #minvapr_mean+
               SporeType,
               phy = tree_trial,
               model = "lambda",
               data=trial)
  
  
  
  modelos<-list(m_all,m_no_life_style,
                m_no_mat_mean,m_no_map_mean,
                m_no_seasonality_t_mean,m_no_seasonality_p_mean,
                m_no_maxsrad_mean,m_no_minvapr_mean)
  names(modelos)<-c("m_all","m_no_life_style",
                "m_no_mat_mean","m_no_map_mean",
                "m_no_seasonality_t_mean","m_no_seasonality_p_mean",
                "m_no_maxsrad_mean","m_no_minvapr_mean")
  
  resultados<-
  do.call("rbind",
  lapply(modelos, function(x){#y<-r2_nakagawa(x)
  data.frame(adj.r.squared=summary(x)$adj.r.squared,
             #R2_fixed=y$R2_marginal,
    df=logLik.phylolm(x)[2],         
    Log_likelihood=logLik.phylolm(x)[1],
    AIC=AIC.phylolm(x))} ))
  
  resultados$dAIC<-resultados$AIC-resultados$AIC[1]
  resultados$Model<-rownames(resultados);rownames(resultados)<-NULL
  resultados<-resultados[,c(6,2,1,3:5)]
  resultados_tax_tree<-resultados;rm(resultados)
  
```

Based on the Li subset
```{r}
resultados_Li_tree
```


Based on the cladograms
```{r}
resultados_tax_tree
```





## Spore aspect ratio
```{r}

### spore aspect ratio

## global model
m1 <- lm(log10.sporeAspectRatio ~ order + simpleFunct + SporeType + 
             alt_mean + mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# # plot to check for nonlinearities
# tmp <- m1$model %>%
#   mutate_if(is.character, as.factor)
# for(i in 2:ncol(tmp)) plot(tmp$log10.sporeAspectRatio ~ tmp[, i], main=names(tmp)[i])

## results for individual variables
m1.var <- vector('list', length=ncol(m1$model))
names(m1.var) <- c('<none>', names(m1$model)[2:ncol(m1$model)])
m1.var[[1]] <- m1$model
for(i in 2:length(m1.var)) m1.var[[i]] <- m1$model[, -i]
m1.aic <- lapply(m1.var, function(x)AIC(m1, lm(log10.sporeAspectRatio ~ ., data=x)))
m1.drp <- data.frame(term = names(m1.var), 
                     Df = sapply(m1.aic, function(x)x[2, 'df']), 
                     AIC = sapply(m1.aic, function(x)x[2, 'AIC']), 
                     AdjR2 = sapply(m1.var, function(x)summary(lm(log10.sporeAspectRatio ~ ., data=x))[['adj.r.squared']])) %>% 
  mutate(type = c('global', rep('subset', n() - 1)), 
         dAIC = AIC - AIC[1], 
         dAdjR2 = AdjR2 - AdjR2[1]) %>% 
  group_by(type) %>% 
  arrange(desc(dAIC), .by_group=TRUE) %>% 
  ungroup() %>% 
  select(term, Df, AIC, dAIC, AdjR2, dAdjR2)

## results for variable groups
# drop taxonomy
m2.tax <- lm(log10.sporeAspectRatio ~ simpleFunct + SporeType + 
               alt_mean + mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop function
m2.fun <- lm(log10.sporeAspectRatio ~ order + SporeType + 
               alt_mean + mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop spore type
m2.typ <- lm(log10.sporeAspectRatio ~ order + simpleFunct + 
               alt_mean + mat_mean + map_mean + seasonality_t_mean + seasonality_p_mean + maxsrad_mean + minvapr_mean, data=trial)
# drop environment
m2.env <- lm(log10.sporeAspectRatio ~ order + simpleFunct + SporeType, data=trial)
# make table
m2.drp <- data.frame(term=c('<none>', 'taxonomy', 'simpleFunct', 'SporeType', 'climate'), 
                     df=AIC(m1, m2.tax, m2.fun, m2.typ, m2.env)[['df']], 
                     AIC=AIC(m1, m2.tax, m2.fun, m2.typ, m2.env)[['AIC']], 
                     AdjR2=sapply(list(m1, m2.tax, m2.fun, m2.typ, m2.env), function(x)summary(x)[['adj.r.squared']]), 
                     row.names=NULL) %>% 
  mutate(type = c('global', rep('subset', n() - 1)), 
         dAIC = AIC - AIC[1], 
         dAdjR2 = AdjR2 - AdjR2[1]) %>% 
  group_by(type) %>% 
  arrange(desc(dAIC), .by_group=TRUE) %>% 
  ungroup() %>% 
  select(term, df, AIC, dAIC, AdjR2, dAdjR2)

# Table 1
dust(m1.drp, caption='Table X. Importance of individual variables in linear models for explaining variation among species in spore shape. Degrees of freedom (df), Akaike\'s Information Criterion (AIC), delta AIC (dAIC), adjusted R^2 (Adj. R2) and delta adjusted R^2 (dAdj. R2) were calculated by dropping terms from a global model containing all terms (indicated in italics).') %>%
  sprinkle_print_method('markdown')  %>%
  sprinkle(rows=1, italic=TRUE) %>%
  sprinkle(cols=c(3, 4), round=1) %>%
  sprinkle(cols=c(5, 6), round=3)

# Table 2
dust(m2.drp, caption='Table X. Importance of variable groupss in linear models for explaining variation among species in spore shape. Degrees of freedom (df), Akaike\'s Information Criterion (AIC), delta AIC (dAIC), adjusted R^2 (Adj. R2) and delta adjusted R^2 (dAdj. R2) were calculated by dropping groups of terms from a global model containing all terms (indicated in italics). ') %>%
  sprinkle_print_method('markdown')  %>%
  sprinkle(rows=1, italic=TRUE) %>%
  sprinkle(cols=c(3, 4), round=1) %>%
  sprinkle(cols=c(5, 6), round=3)

```


# Figures

## Figure 1 (further modified in inkspace)

### Figure 1 Phylogenetic Tree

```{r}
library(ape)
library(ggtree)

#devtools::install_github("ropenscilabs/datastorr")
#devtools::install_github("wcornwell/taxonlookup")

library(taxonlookup)

# phylogeny from https://www.nature.com/articles/s41467-018-07849-9
tt<-read.nexus("phylo/Lutzoni_fungi_timetree.nex")
vcapply<-taxonlookup:::vcapply

split<-function(str){
  str_split <- strsplit(str, "[_ ]+")
  vcapply(str_split, "[[", 2L)
}

#generate genus in order lookup table
#fungal_genera<-split(nnn)[1:197]
#write_csv(tibble(genus=fungal_genera),"phylo/genus_names.csv")

#read in genus in order lookup
orders<-read_csv("phylo/orders_phylo.csv")
orders %>% group_by(order) %>%
  summarize(genus=genus[1]) -> only_one_genera_per_order

#getting a tree with one tip per order
nnn<-tt$tip.label
tt$tip.label<-split(nnn)
dropped.tree<-drop.tip(tt,tt$tip.label[which(!tt$tip.label %in% only_one_genera_per_order$genus)])
dup.tips<-names(which(table(dropped.tree$tip.label)>1))
tips<-dropped.tree$tip.label 
dropped.tree2<-drop.tip(dropped.tree,which(duplicated(tips)))
dropped.tree2$tip.label<-orders$order[match(dropped.tree2$tip.label,orders$genus)]



#read in spore data and calculate size and shape variables
#sdf<-read_csv("output/Spore_Database_Fungi.csv")
matched_data<-filter(AllFungi,order%in%dropped.tree2$tip.label)
matched_data$log.length<-log10(matched_data$spore_length)
#matched_data$log_spore_area<-log10(matched_data$spore_length*matched_data$spore_width * pi /4)
matched_data$log_spore_volume<-log10((matched_data$spore_width^2)*matched_data$spore_length*(pi/6))
matched_data$length_divided_by_width<-log10(matched_data$spore_length/matched_data$spore_width)


#check tree
dropped.tree2$tip.label %in% matched_data$order

#relabel to ggtree convention
matched_data$id <- matched_data$order

#drop not assigned taxa; there are multiple spots on tree with non assigned taxa
matched_data<-filter(matched_data,order!="Not assigned")
dropped.tree2<-drop.tip(dropped.tree2,"Not assigned")

#drop small spore size categories for plotting
match2<-filter(matched_data,!SporeName %in% c("bulbil","papulospore"))

#select only necessary data
#temp<-select(match2,id,log_spore_area,SporeName,length_divided_by_width)
temp<-select(match2,
             id, phylum,log_spore_volume, SporeName, length_divided_by_width)

temp$SporeType<-NA
temp$SporeType[which(temp$SporeName=="Ascospores"|temp$SporeName=="Basidiospores")]<-"Meiospores"
temp$SporeType[which(temp$SporeName=="Conidia"|temp$SporeName=="Sporangiospores"|temp$SporeName=="Chlamydospores")]<-"Mitospores"
temp$SporeType[which(temp$SporeName=="Azygospores")]<-"Multinucleate asexual spores"
temp$SporeType[which(temp$SporeName=="Zygospores")]<-"Multinucleate sexual spores"
temp$SporeType[which(temp$SporeName=="Teliospores")]<-"Multinucleate sexual spores"


# summarize data # My modification
sumbox<-temp %>%
  group_by(id, phylum,SporeType) %>%
  summarise(
    n=n(),
    meanV=mean(log_spore_volume),
    meanQ=mean(length_divided_by_width),
    sdV=sd(log_spore_volume),
    sdQ=sd(length_divided_by_width)
  ) %>%
  mutate(seV=sdV/sqrt(n)) %>%
  mutate(seQ=sdQ/sqrt(n))

#sumbox[sumbox$n==1,][,c("meanV","meanQ","sdV","sdQ","seV","seQ")]<-NA
#Orders for which we only have 1 species
just_one<-
  temp %>% 
  group_by(id) %>% 
  count() %>% 
  filter(n==1)
#These are the Basidiobolales and the Dimargitales

dropped.tree2<-drop.tip(dropped.tree2,dropped.tree2$tip.label[match(just_one$id, 
                                                                    dropped.tree2$tip.label)])

sumbox<-sumbox[-which(sumbox$n==1),]

#my changes:
grouped_tree <- groupClade(dropped.tree2, .node=c(72, 104,65), group_name = "Phyla")


# c("Ascomycota"          = "#001399", # This is blue
#   "Basidiomycota"       = "#C40233",#This is red 
#   "Glomeromycota"       = "#19461A",
#   "Zygomycota"          = "#19461A")#,This is greenish

#My modifications
pltre_neutral <- ggtree(grouped_tree, aes( color=Phyla))+ 
  geom_tiplab(size = 2)+ 
  theme_tree2()+
  ggplot2::xlim(0, 800)+
  scale_color_manual(
    values=c("black","#001399","#C40233","#19461A")
  ) +
  theme(legend.position = "none")

pltre_neutral
```

```{r}
paso1<-
AllFungi[which(AllFungi$order%in%grouped_tree$tip.label),c("phylum","order")]

paso1<-paso1[-which(duplicated(paso1$order)),]

paso2<-
left_join(
paso1,
To_Analysis[,c("order","Life_style")] %>% 
  mutate(Life_style = recode(Life_style, Plant = "Facultative association"))
)

paso2$Life_style<-gsub(" ","_",paso2$Life_style)

paso3<-
paso2[-which(duplicated(paste(paso2$order,paso2$Life_style))),c("order","Life_style")]


# paso3<-
# do.call("rbind",
# lapply(
# split(paso2,paso2$order
# ),function(x){
# data.frame(order=unique(x$order),
#            Life_style=paste0(unique(x$Life_style), collapse = " "))
# }
# )
# )

```


There are 8 orders in the three for which i did not get lifestyle data: 
"Lulworthiales"      "Platygloeales"      "Sporidiobolales"    "Heterogastridiales" "Kickxellales"      
"Harpellales"        "Endogonales"      "Zoopagales"

Thus I look at diferent sources to get general descriptions of the ecology of these orders without going to every species. 

Zoopagales: obligate parasites, according to The Mycota Systematics and evolution part a vii second edition editted by karl esser on page 235.

Lulworthiales:members of this order degrade wood and marsh plants in marine and estuarine environments, some species are also included that are obligate parasites of algae according to The Mycota Systematics and evolution part b vii second edition editted by karl esser on page 68

Platyloeales: parasties of mosses and ferns according to The Mycota Systematics and evolution part a vii second edition editted by karl esser on page 289

Sporidiobolales: epiphytes of animals, and common inhabitants of soils and extreme environments according to Hector Urbina & M. Catherine Aime (2018) A closer look at Sporidiobolales:
Ubiquitous microbial community members of plant and food biospheres, Mycologia, 110:1, 79-92

Heterogastridales: are mycoparasites according to The Mycota Systematics and evolution part a vii second edition editted by karl esser on page 287-288

Kickxellales: have only been isoted from soil or dung accordign to The Mycota Systematics and evolution part a vii second edition editted by karl esser on page 232

Harpelalles: all found as symbionts in the hind or hindgut fo inmmature aquatic insects according to The Mycota Systematics and evolution part a vii second edition editted by karl esser on page 231

Endogonales: are saprotrophs or ectomycorrhizal according to The Mycota Systematics and evolution part a vii second edition editted by karl esser on page 233


Based on this information I am assigning these entries
```{r}
paso3$Life_style[which(paso3$order=="Zoopagales")]<-"Obligate_association"
paso3$Life_style[which(paso3$order=="Platygloeales")]<-"Obligate_association"
paso3$Life_style[which(paso3$order=="Heterogastridiales")]<-"Obligate_association"
paso3$Life_style[which(paso3$order=="Kickxellales")]<-"AFree_living"
paso3$Life_style[which(paso3$order=="Harpellales")]<-"Obligate_association"

paso3$Life_style[which(paso3$order=="Lulworthiales")]<-"AFree_living"
paso3<-rbind(paso3,data.frame(order="Lulworthiales",Life_style="Obligate_association"))

paso3$Life_style[which(paso3$order=="Sporidiobolales")]<-"AFree_living"
paso3<-rbind(paso3,data.frame(order="Sporidiobolales",Life_style="Facultative_association"))

paso3$Life_style[which(paso3$order=="Endogonales")]<-"AFree_living"
paso3<-rbind(paso3,data.frame(order="Endogonales",Life_style="Obligate_association"))

#Ordering the orders as they appear in the tree
d = fortify(dropped.tree2)
d = subset(d, isTip)
#with(d, label[order(y, decreasing=T)])

# reorder id factor
paso3$order<-factor(paso3$order, levels = with(d, label[order(y, decreasing=F)]))

paso3<-paso3[order(factor(paso3$order,levels=levels(paso3$order))),]

#Because we are considering opportunistic pathogens as not really adapted to symbbiosis
paso3$Life_style<-gsub("B_Opportunistic_association","AFree_living",paso3$Life_style)

paso3$value<-1
paso3$value[grep("Facultative",paso3$Life_style)]<-2
paso3$value[grep("Obligate",paso3$Life_style)]<-3


paso3 %>% 
  mutate(Life_style=factor(Life_style,
                              levels =c("Obligate_association","Facultative_association","AFree_living"))) %>% 
  ggplot()+
  aes(x=value,y=order,shape=Life_style)+
  geom_point(size=5)+
  scale_x_continuous(breaks = c(1,2,3))+
  theme(panel.background = element_blank(),
        axis.title = element_blank(),
        axis.text.x = element_blank(),
        legend.text = element_text(size = 0.5)
        )
  


#Just to calculate how many species are we not including because of these orders
orders_in_tree_no_lifestyle_info<-c("Lulworthiales","Platygloeales","Sporidiobolales",    "Heterogastridiales","Kickxellales","Harpellales","Endogonales","Zoopagales")

orders_in_tree_no_lifestyle_info2<-c("Platygloeales",    "Heterogastridiales","Kickxellales","Harpellales","Zoopagales")

unique(
AllFungi$names_to_use[which(AllFungi$order%in%orders_in_tree_no_lifestyle_info)])

unique(
AllFungi$names_to_use[which(AllFungi$order%in%orders_in_tree_no_lifestyle_info2)])

```




### Figure 1 Comparing spore size with that of seeds and eggs

```{r}
rbind(
  AllFungi%>%
        group_by(phylum,names_to_use,SporeName,Specific_sporeName)%>%
      summarise_at(c("spore_length","spore_width","SporeArea","SporeVolume","Q_ratio"),mean) %>% 
                          select(names_to_use,SporeVolume)%>%
                          rename(offsrpingSize=SporeVolume)%>%
                          #filter(good.names!="Glomus_tenue")%>%
                          mutate(Taxa="Fungi")%>%
                          mutate(offsrpingSize=offsrpingSize/((10000)^3)),#%>%
                          #mutate(Organism="Microorganism"),
                        
                        kewData%>%
                          select(species,value)%>%
                          rename(names_to_use=species, offsrpingSize=value)%>%
                          mutate(Taxa="Plants"),#%>%
                          #mutate(Organism="Macroorganism"),
                        BirdEgg%>%
                          select(Species,Volume)%>%
                          rename(names_to_use=Species,offsrpingSize=Volume)%>%
                          mutate(Taxa="Birds"))%>%
                      #mutate(Taxa=factor(Taxa,levels =c("Birds","Plants","Fungi"))) %>% 
                      mutate(Taxa=factor(Taxa,levels = c("Fungi","Plants","Birds"))) %>% #For the coord flip version is better in this reverse order
                          #mutate(Organism="Macroorganism"))%>%
                    
                    ggplot()+
                    aes(Taxa,offsrpingSize)+
                    geom_jitter(alpha=0.9,size=4,width = 0.3,color="#FD8D3C")+
  #fe8d3ce6
                    geom_violin(alpha=0.5,
                          lwd=1,
                          fill=NA)+
                    #facet_grid(. ~ Organism, scales = "free")+
                    scale_y_log10(labels = scales::trans_format("log10", scales::math_format(10^.x)))+
                    labs(x="Offspring structure",y=expression("Offspring size as volume ("*c*"m³)"))+
                    #scale_color_brewer(palette="Accent")+
                    scale_x_discrete(labels=c("Birds" = "Birds\n(eggs)", "Plants"="Plants\n(seeds)","Fungi"="Fungi\n(spores)"))+
  
                      theme(title = element_text(size = 18),
                      panel.background = element_blank(),
                      panel.grid.major = element_line(size = 0.5, linetype = 'longdash',
                                                      colour = "gray"),
                      # panel.grid.minor = element_line(size = 0.5, linetype = 'longdash',
                      #                                 colour = "gray"),
                      axis.title.x=element_blank(),
                      axis.text.x = element_text(size = 20),#,angle = 45,hjust = 1),
                      axis.text.y = element_blank(),
                      #axis.title.x = element_text(size = 20),
                      axis.title.y = element_blank(),
                      legend.position = "none")+
  coord_flip()

```

### Figure 1 Heatmaps

```{r}
# get list of taxa in order in which they occur on tree
d = fortify(dropped.tree2)
d = subset(d, isTip)
with(d, label[order(y, decreasing=T)])

# reorder id factor
sumbox$id<-factor(sumbox$id, levels = with(d, label[order(y, decreasing=F)]))


datos<-sumbox[order(factor(sumbox$id,levels=levels(sumbox$id))),]

datos$Vol_CV<-datos$sdV/datos$meanV

datos$Q_CV<-datos$sdQ/(datos$meanQ+1)#Adding 1 is a quick solution I am not sure about it

#rownames(datos)<-datos$id
datos<-pivot_wider(datos[,c("id","SporeType","meanV","meanQ","Vol_CV","Q_CV")],
                   names_from = SporeType,values_from = c(meanV,Vol_CV,meanQ,Q_CV))


datos1<-as.matrix(datos[,c( "meanV_Meiospores",#"sdV_Meiospores",
                            "meanV_Mitospores",#"sdV_Mitospores",
                            "meanV_Multinucleate sexual spores",
                            "meanV_Multinucleate asexual spores")])#"sdV_Multinucleate asexual spores",
                            #,"sdV_Multinucleate sexual spores"#,
                                                        #)])
rownames(datos1)<-datos$id


datos2<-as.matrix(datos[,c( "Vol_CV_Meiospores",
                            "Vol_CV_Mitospores",
                            "Vol_CV_Multinucleate sexual spores",
                            "Vol_CV_Multinucleate asexual spores")])
rownames(datos2)<-datos$id
##
datos3<-as.matrix(datos[,c( "meanQ_Meiospores",
                            "meanQ_Mitospores",
                            "meanQ_Multinucleate sexual spores",
                            "meanQ_Multinucleate asexual spores")])
rownames(datos3)<-datos$id

datos4<-as.matrix(datos[,c( "Q_CV_Meiospores",
                            "Q_CV_Mitospores",
                            "Q_CV_Multinucleate sexual spores",
                            "Q_CV_Multinucleate asexual spores")])
rownames(datos4)<-datos$id
```


```{r}
library(RColorBrewer)
coul<-colorRampPalette(brewer.pal(9, "Blues"))(100)
#Volume
heatmap(datos1,Colv = NA, Rowv = NA,cexCol = 0.5,cexRow = 0.5,scale = "none",
        col=hcl.colors(12, "YlOrRd", rev = TRUE))
legend(x="bottomright", legend=c("a", "b", "c","d","e"), 
       fill=hcl.colors(5, "YlOrRd", rev = TRUE)[5:1])

heatmap(datos2,Colv = NA, Rowv = NA,cexCol = 0.5,cexRow = 0.5,scale = "none",
        col=coul)
```


```{r}
#Aspect ratio
heatmap(datos3,Colv = NA, Rowv = NA,cexCol = 0.5,cexRow = 0.5,scale = "none",
        col=hcl.colors(12, "Greens", rev = TRUE))
legend(x="bottomright", legend=c("a", "b", "c","d","e"), 
       fill=hcl.colors(5, "Greens", rev = TRUE)[5:1])

heatmap(datos4,Colv = NA, Rowv = NA,cexCol = 0.5,cexRow = 0.5,scale = "none",
        col=coul)
legend(x="bottomright", legend=c("a", "b", "c","d","e"), 
       fill=colorRampPalette(brewer.pal(9, "Blues"))(5)[5:1])
```


## Figure 1 violin plots spore size and shape across phyla and spore type


```{r}

AllFungi%>%
          filter(phylum!="Chytridiomycota")%>%#Filtering only because there are so few
          filter(!(phylum=="Ascomycota"&SporeName=="Basidiospores"))%>% #weird cases
          filter(!(phylum=="Ascomycota"&SporeName=="Teliospores"))%>% #weird  cases
          filter(!(phylum=="Basidiomycota"&SporeName=="Ascospores"))%>% #weird  cases
          filter(SporeName!="papulospore")%>%#Filtering only because there are so few
          filter(SporeName!="bulbil")%>%#Filtering only because there are so few
          
          group_by(phylum,SporeType,SporeName,Specific_sporeName,names_to_use)%>%
          summarise_at(c("SporeVolume","Q_ratio"),mean)%>%
          
          ungroup() %>% 
          
          mutate(phylum_type=paste(phylum,SporeType,sep = " ")) %>% 
          mutate(phylum_type=factor(phylum_type,levels = c("Ascomycota Meiospores","Ascomycota Mitospores","Basidiomycota Meiospores","Basidiomycota Mitospores","Basidiomycota Multinucleate sexual spores","Zygomycetous_fungi Mitospores","Zygomycetous_fungi Multinucleate asexual spores","Zygomycetous_fungi Multinucleate sexual spores","Glomeromycota Multinucleate asexual spores"))) %>% 
          
          select(phylum_type,names_to_use,SporeVolume,Q_ratio) %>% 
          group_by(names_to_use) %>% 
          gather(key=variable,
               value=value,SporeVolume:Q_ratio)%>%
  
          group_by(phylum_type,variable) %>% 
          mutate(phylum_mean=mean(log10(value))) %>%
          ungroup() %>% 
        
          ggplot()+
          aes(phylum_type,value,fill=phylum_mean)+
          scale_fill_continuous_sequential(palette="YlOrRd")+
          geom_violin(scale = "width",alpha=0.8, draw_quantiles=c(0.25, 0.5, 0.75))+
          facet_grid(variable~., scales = "free_y")+
          scale_y_log10(breaks=c(10^0,10^1,10^2,10^3,10^4,10^5,10^6,10^7),
                        labels = scales::trans_format("log10", scales::math_format(10^.x)))+
          labs(y=expression("Spore size as Volume ("*mu*"m³)"))+
  
        theme(title = element_text(size = 18),
        panel.background = element_blank(),
        panel.grid.major.y = element_line(size = 0.5, linetype = 'longdash',
                                        colour = "gray"),
        
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.x = element_blank(),
        axis.text.y = element_text(size = 15),
        strip.text = element_blank(),
        legend.position = "none")

```




## Figure 2 and Table S7 (colors were modified manually in inkscape)

```{r}
library(ggridges)
To_Analysis1<-To_Analysis
To_Analysis1<-To_Analysis1[-which(To_Analysis1$simpleFunct=="Animal"),]
To_Analysis1<-To_Analysis1[-which(To_Analysis1$simpleFunct=="Fungi"),]
To_Analysis_c<-To_Analysis1;rm(To_Analysis1)

To_Analysis_c$phylum[To_Analysis_c$phylum=="\"Zygomycetous fungi\""]<-"Zygomycetous fungi"

To_Analysis_c<-To_Analysis_c %>% 
  mutate(simpleFunct=factor(simpleFunct,levels = c("A_Saprotroph","Human","Plant Endophyte","Plant necrotroph","Plant Pathogen undefined","Plant Ectomycorrhizal","Insect","Lichen","Plant Pathogen Smut","Plant Pathogen Mildew","Plant Pathogen Rust")))

To_Analysis_c$Life_style<-gsub(" ","_",To_Analysis_c$Life_style)
```

Figure 2 Obtaining the mean meiospores size for saprotrophic fungi
```{r}
tapply(log10(To_Analysis_c$SporeVolume[which(To_Analysis_c$SporeType=="Meiospores")]),
        list(To_Analysis_c$simpleFunct[which(To_Analysis_c$SporeType=="Meiospores")],
             To_Analysis_c$phylum[which(To_Analysis_c$SporeType=="Meiospores")]),
       mean)
```


Figure 2 Obtaining the mean mitospore size for saprotrophic fungi
```{r}
tapply(log10(To_Analysis_c$SporeVolume[which(To_Analysis_c$SporeType=="Meiospores")]),
        list(To_Analysis_c$simpleFunct[which(To_Analysis_c$SporeType=="Meiospores")],
             To_Analysis_c$phylum[which(To_Analysis_c$SporeType=="Meiospores")]),
       length)
```



Table S7. Number of unique species (accepted species names) for which we obtained per phylum, spore type and functional group.
```{r}

contando_grupos<-
bind_rows(
To_Analysis_c %>% 
  filter(SporeType=="Meiospores") %>% 
  filter(!(SporeName=="Ascospores"&simpleFunct=="Plant Ectomycorrhizal")) %>% 
  filter(!(SporeName=="Ascospores"&simpleFunct=="Plant Pathogen Smut")) %>% 
  filter(!(phylum=="Basidiomycota"&simpleFunct=="Human")) %>% 
  filter(!(phylum=="Basidiomycota"&SporeVolume>90000)) %>% #One species (which I checked) has a very high volume, because ridges function extend the tail with a color because of only this point I am removing it
  mutate(Life_style = recode(Life_style, Plant = "Facultative_association")) %>% 
  mutate(simpleFunct=factor(simpleFunct,levels = c("Saprotroph","Human","Plant Endophyte","Plant necrotroph","Plant Pathogen undefined","Plant Ectomycorrhizal","Insect","Lichen","Plant Pathogen Smut","Plant Pathogen Mildew","Plant Pathogen Rust"))) %>% 
  group_by(phylum,simpleFunct) %>% 
  count() %>% 
  mutate(Spore_type="Meiospores"),

To_Analysis_c %>% 
  filter(SporeType=="Mitospores") %>% 
  filter(!(phylum=="Ascomycota"&simpleFunct=="Plant Pathogen Smut")) %>% #There is only 1 and it is actually a "false smut"
  filter(!(phylum=="Basidiomycota"&simpleFunct=="Human")) %>% #Thre is only 1
  filter(!(phylum=="Basidiomycota"&simpleFunct=="Plant Ectomycorrhizal")) %>% #There are only 3
  filter(!(phylum=="Basidiomycota"&simpleFunct=="Plant Pathogen undefined")) %>% #There are only 3
  mutate(Life_style = recode(Life_style, Plant = "Facultative_association")) %>%
  group_by(phylum,simpleFunct) %>% 
  count() %>% 
  mutate(Spore_type="Mitospores"))

write.csv(contando_grupos,"contando_grupos.csv",row.names = F)

```


Figure 2. Spore size along different lifestyles

Upper panel
```{r}
To_Analysis_c %>% 
  #filter(SporeType=="Meiospores"|SporeType=="Mitospores") %>% 
  filter(SporeType=="Meiospores") %>% 
  filter(!(SporeName=="Ascospores"&simpleFunct=="Plant Ectomycorrhizal")) %>% 
  filter(!(SporeName=="Ascospores"&simpleFunct=="Plant Pathogen Smut")) %>% 
  filter(!(phylum=="Basidiomycota"&simpleFunct=="Human")) %>% 
  filter(!(phylum=="Basidiomycota"&SporeVolume>90000)) %>% #One species (which I checked) has a very high volume, because ridges function extend the tail with a color because of only this point I am removing it
  mutate(Life_style = recode(Life_style, Plant = "Facultative_association")) %>% 
  mutate(simpleFunct=factor(simpleFunct,levels = c("Saprotroph","Human","Plant Endophyte","Plant necrotroph","Plant Pathogen undefined","Plant Ectomycorrhizal","Insect","Lichen","Plant Pathogen Smut","Plant Pathogen Mildew","Plant Pathogen Rust"))) %>%
  
  ggplot()+
  aes(y=simpleFunct,
      x=SporeVolume,
      
  )+
geom_density_ridges(aes(fill = Life_style,point_alpha=0.05),jittered_points = TRUE, #position = "raincloud",
     alpha=0.9,scale = 2,rel_min_height = 0.01)+
  #scale_fill_brewer(palette="YlOrRd")+
  scale_fill_manual(values= c("#FFFFB2","#FECC5C","#FD8D3C", "#E31A1C"))+
  scale_x_log10(breaks=c(10^0,10^1,10^2,10^3,10^4,10^5,10^6,10^7),
                labels = scales::trans_format("log10", scales::math_format(10^.x)))+
  facet_wrap(.~phylum, strip.position =  "top",nrow = 1,drop=FALSE,scales = "free_x")+
  
    theme(title = element_text(size = 18),
        panel.background = element_blank(),
        
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line(size = 0.25, linetype = 'longdash',
                                        colour = "gray"),
        axis.title.y=element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(size = 25),#,angle = 45,hjust = 1),
        axis.text.y = element_text(size = 5),
        strip.text.x = element_blank(),
        strip.text.y = element_blank(),
        legend.position = "bottom",
        panel.spacing = unit(-3, 'inches'))+
#   #For ascomycota
   geom_vline(aes(xintercept=10^2.64),color="red",linetype="longdash")+
 
#   #For basidiomycota
   geom_vline(aes(xintercept=10^2.06),color="blue",linetype="longdash")
 


```


Figure 2 Lower panel
```{r}

To_Analysis_c %>% 
  #filter(SporeType=="Meiospores"|SporeType=="Mitospores") %>% 
  filter(SporeType=="Mitospores") %>% 
  
  #filter(!(simpleFunct=="Plant Ectomycorrhizal")) %>% 
  filter(!(phylum=="Ascomycota"&simpleFunct=="Plant Pathogen Smut")) %>% #There is only 1 and it is actually a "false smut"
  filter(!(phylum=="Basidiomycota"&simpleFunct=="Human")) %>% #Thre is only 1
  filter(!(phylum=="Basidiomycota"&simpleFunct=="Plant Ectomycorrhizal")) %>% #There are only 3
  filter(!(phylum=="Basidiomycota"&simpleFunct=="Plant Pathogen undefined")) %>% #There are only 3
  mutate(Life_style = recode(Life_style, Plant = "Facultative_association")) %>%   
  #filter(phylum=="Ascomycota") %>% 
  #filter(phylum=="Basidiomycota") %>% 
  ggplot()+
  aes(y=simpleFunct,
      x=SporeVolume,
      
  )+
geom_density_ridges(aes(fill = Life_style,point_alpha=0.05),jittered_points = TRUE, #position = "raincloud",
     alpha=0.9,scale = 2,rel_min_height = 0.01)+
  scale_fill_brewer(palette="YlOrRd")+
  scale_x_log10(breaks=c(10^0,10^1,10^2,10^3,10^4,10^5,10^6,10^7),
                labels = scales::trans_format("log10", scales::math_format(10^.x)))+
  facet_wrap(.~phylum, strip.position =  "top",nrow = 1,drop=FALSE)+
  
    theme(title = element_text(size = 18),
        panel.background = element_blank(),
        
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line(size = 0.25, linetype = 'longdash',
                                        colour = "gray"),
        axis.title.y=element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(size = 25),#,angle = 45,hjust = 1),
        axis.text.y = element_text(size = 5),
        strip.text.x = element_blank(),
        strip.text.y = element_blank(),
        legend.position = "bottom",
        panel.spacing = unit(-3, 'inches'))+
#   #For ascomycota
   geom_vline(aes(xintercept=10^2.17),color="red",linetype="longdash")+
 
#   #For basidiomycota
   geom_vline(aes(xintercept=10^2.6),color="green",linetype="longdash")+
 
# #For zygomycota
   geom_vline(aes(xintercept=10^2.44),color="blue",linetype="longdash")
```




Trying sth

```{r}
tapply(log10(To_Analysis_c$SporeVolume[which(To_Analysis_c$SporeType=="Meiospores")]),
        list(To_Analysis_c$simpleFunct[which(To_Analysis_c$SporeType=="Meiospores")],
             To_Analysis_c$phylum[which(To_Analysis_c$SporeType=="Meiospores")]),
       mean)
```


```{r}
pdf("Figures\\Alternative_2_meiospores.pdf",
    width = 37,
    height = 14)

To_Analysis_c %>% 
  #filter(SporeType=="Meiospores"|SporeType=="Mitospores") %>% 
  filter(SporeType=="Meiospores") %>% 
  #filter(!(SporeName=="Ascospores"&simpleFunct=="Plant Ectomycorrhizal")) %>% 
  #filter(!(SporeName=="Ascospores"&simpleFunct=="Plant Pathogen Smut")) %>% 
  filter(!(phylum=="Basidiomycota"&simpleFunct=="Human")) %>% 
  filter(!(phylum=="Basidiomycota"&SporeVolume>90000)) %>% #One species (which I checked) has a very high volume, because ridges function extend the tail with a color because of only this point I am removing it
  mutate(Life_style = recode(Life_style, Plant = "Facultative_association")) %>% 
  mutate(simpleFunct=factor(simpleFunct,levels = c("Saprotroph","Human","Plant Endophyte","Plant necrotroph","Plant Pathogen undefined","Plant Ectomycorrhizal","Insect","Lichen","Plant Pathogen Smut","Plant Pathogen Mildew","Plant Pathogen Rust"))) %>%
  
  ggplot()+
  aes(y=Life_style,
      x=SporeVolume,
      
  )+
geom_density_ridges(aes(fill = Life_style,point_alpha=0.05),jittered_points = TRUE, #position = "raincloud",
     alpha=0.9,scale = 2,rel_min_height = 0.01)+
  #scale_fill_brewer(palette="YlOrRd")+
  scale_fill_manual(values= c("#FFFFB2","#FECC5C","#FD8D3C", "#E31A1C"))+
  scale_x_log10(breaks=c(10^0,10^1,10^2,10^3,10^4,10^5,10^6,10^7),
                labels = scales::trans_format("log10", scales::math_format(10^.x)))+
  facet_wrap(.~phylum, strip.position =  "top",nrow = 1,drop=FALSE,scales = "free_x")+
  
    theme(title = element_text(size = 18),
        panel.background = element_blank(),
        
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line(size = 0.25, linetype = 'longdash',
                                        colour = "gray"),
        axis.title.y=element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(size = 25),#,angle = 45,hjust = 1),
        axis.text.y = element_text(size = 5),
        strip.text.x = element_blank(),
        strip.text.y = element_blank(),
        legend.position = "bottom",
        panel.spacing = unit(-3, 'inches'))+
#   #For ascomycota
   geom_vline(aes(xintercept=10^2.64),color="red",linetype="longdash")+
 
#   #For basidiomycota
   geom_vline(aes(xintercept=10^2.06),color="blue",linetype="longdash")

dev.off()
 
```


```{r}
pdf("Figures\\Alternative_2_mitospores.pdf",
    width = 37,
    height = 14)

To_Analysis_c %>% 
  #filter(SporeType=="Meiospores"|SporeType=="Mitospores") %>% 
  filter(SporeType=="Mitospores") %>% 
  #filter(!(SporeName=="Ascospores"&simpleFunct=="Plant Ectomycorrhizal")) %>% 
  #filter(!(SporeName=="Ascospores"&simpleFunct=="Plant Pathogen Smut")) %>% 
  filter(!(phylum=="Basidiomycota"&simpleFunct=="Human")) %>% 
  #filter(!(phylum=="Basidiomycota"&SporeVolume>90000)) %>% #One species (which I checked) has a very high volume, because ridges function extend the tail with a color because of only this point I am removing it
  mutate(Life_style = recode(Life_style, Plant = "Facultative_association")) %>% 
  mutate(simpleFunct=factor(simpleFunct,levels = c("Saprotroph","Human","Plant Endophyte","Plant necrotroph","Plant Pathogen undefined","Plant Ectomycorrhizal","Insect","Lichen","Plant Pathogen Smut","Plant Pathogen Mildew","Plant Pathogen Rust"))) %>%
  
  ggplot()+
  aes(y=Life_style,
      x=SporeVolume,
      
  )+
geom_density_ridges(aes(fill = Life_style,point_alpha=0.05),jittered_points = TRUE, #position = "raincloud",
     alpha=0.9,scale = 2,rel_min_height = 0.01)+
  #scale_fill_brewer(palette="YlOrRd")+
  scale_fill_manual(values= c("#FFFFB2","#FECC5C","#FD8D3C", "#E31A1C"))+
  scale_x_log10(breaks=c(10^0,10^1,10^2,10^3,10^4,10^5,10^6,10^7),
                labels = scales::trans_format("log10", scales::math_format(10^.x)))+
  facet_wrap(.~phylum, strip.position =  "top",nrow = 1,drop=FALSE,scales = "free_x")+
  
    theme(title = element_text(size = 18),
        panel.background = element_blank(),
        
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line(size = 0.25, linetype = 'longdash',
                                        colour = "gray"),
        axis.title.y=element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(size = 25),#,angle = 45,hjust = 1),
        axis.text.y = element_text(size = 5),
        strip.text.x = element_blank(),
        strip.text.y = element_blank(),
        legend.position = "bottom",
        panel.spacing = unit(-3, 'inches'))+
#   #For ascomycota
   geom_vline(aes(xintercept=10^2.17),color="red",linetype="longdash")+
 
#   #For basidiomycota
   geom_vline(aes(xintercept=10^2.6),color="green",linetype="longdash")+
 
# #For zygomycota
   geom_vline(aes(xintercept=10^2.44),color="blue",linetype="longdash")

dev.off()
```


##Figure 3

## Spore volume

```{r, eval=FALSE}

library(brms)

# m1 <- lmer(car::logit(geo_area_prop) ~ log10.sporeVolume * Life_style + 
#                  (1|order) + (1|Primers.name) + (1|SporeType), data=temp)
# m2 <- lmer(car::logit(geo_area_prop) ~ log10.sporeVolume * Life_style + 
#                  (1|order/Life_style) + (1|Primers.name) + (1|SporeType), data=temp)
# anova(m2, m1)

# fit1.ext <- brm(formula = car::logit(geo_extent_prop) ~ log10.sporeVolume * Life_style + 
#                   (1|order) + (1|Primers.name) + (1|SporeType),
#                 data=drop_na(select(temp, geo_extent_prop, log10.sporeVolume, Life_style, order, Primers.name, SporeType)),
#                 family = gaussian(),
#                 prior = c(set_prior("normal(0,5)", class = "b"),
#                           set_prior("student_t(10,0,1)", class = "sd")),
#                 warmup = 1000, iter = 2000, chains = 4,
#                 control = list(adapt_delta = 0.995))
# fit1.ext.loo <- loo(fit1.ext)
# fit1.area <- brm(formula = car::logit(geo_area_prop) ~ log10.sporeVolume * Life_style + 
#                   (1|order) + (1|Primers.name) + (1|SporeType),
#                 data=drop_na(select(temp, geo_area_prop, log10.sporeVolume, Life_style, order, Primers.name, SporeType)),
#                 family = gaussian(),
#                 prior = c(set_prior("normal(0,5)", class = "b"),
#                           set_prior("student_t(10,0,1)", class = "sd")),
#                 warmup = 1000, iter = 2000, chains = 4,
#                 control = list(adapt_delta = 0.995))
# fit1.area.loo <- loo(fit1.area)

# save(fit1.ext, fit1.ext.loo, fit1.area, fit1.area.loo, file='output/brmsoutput_sporeVolume.RData')

```


```{r, echo=FALSE}

library(ggridges)
library(patchwork)
temp<-left_join(read_csv('output/df_species_byPrimerSet.csv') %>%
                   mutate(names_to_use = gsub("_", " ", Species)) %>%
                   select(names_to_use, host.assoc, Primers.name, log10.sporeVolume, log10.sporeAspectRatio, starts_with('geo_'), ends_with('_sd'), n_samples_env),
                 To_Analysis[,c("names_to_use","SporeType","simpleFunct","Life_style",
                                "phylum","class","order")]) %>% 
  filter(!order == 'Not assigned') %>% 
  mutate(Life_style = recode(Life_style, Plant = "Facultative association", 
                             `B_Opportunistic association` = "AFree living"))


load('output/brmsoutput_sporeVolume.RData')

### confidence bands for slope

# ext
slp.a <- c(attr(fit1.ext$fit, 'sim')$samples[[1]][['b_log10.sporeVolume']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[2]][['b_log10.sporeVolume']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[3]][['b_log10.sporeVolume']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[4]][['b_log10.sporeVolume']][1001:2000])
conf.95.a <- slp.a > quantile(slp.a, prob=0.025) & slp.a < quantile(slp.a, prob=0.975)
conf.50.a <- slp.a > quantile(slp.a, prob=0.25) & slp.a < quantile(slp.a, prob=0.75)
# slp.b <- c(attr(fit1.ext$fit, 'sim')$samples[[1]][['b_log10.sporeVolume:Life_styleB_Opportunisticassociation']][1001:2000], 
#            attr(fit1.ext$fit, 'sim')$samples[[2]][['b_log10.sporeVolume:Life_styleB_Opportunisticassociation']][1001:2000], 
#            attr(fit1.ext$fit, 'sim')$samples[[3]][['b_log10.sporeVolume:Life_styleB_Opportunisticassociation']][1001:2000], 
#            attr(fit1.ext$fit, 'sim')$samples[[4]][['b_log10.sporeVolume:Life_styleB_Opportunisticassociation']][1001:2000])
# conf.95.b <- slp.b > quantile(slp.b, prob=0.025) & slp.b < quantile(slp.b, prob=0.975)
# conf.50.b <- slp.b > quantile(slp.b, prob=0.25) & slp.b < quantile(slp.b, prob=0.75)
slp.c <- c(attr(fit1.ext$fit, 'sim')$samples[[1]][['b_log10.sporeVolume:Life_styleFacultativeassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[2]][['b_log10.sporeVolume:Life_styleFacultativeassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[3]][['b_log10.sporeVolume:Life_styleFacultativeassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[4]][['b_log10.sporeVolume:Life_styleFacultativeassociation']][1001:2000])
conf.95.c <- slp.c > quantile(slp.c, prob=0.025) & slp.c < quantile(slp.c, prob=0.975)
conf.50.c <- slp.c > quantile(slp.c, prob=0.25) & slp.c < quantile(slp.c, prob=0.75)
slp.d <- c(attr(fit1.ext$fit, 'sim')$samples[[1]][['b_log10.sporeVolume:Life_styleObligateassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[2]][['b_log10.sporeVolume:Life_styleObligateassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[3]][['b_log10.sporeVolume:Life_styleObligateassociation']][1001:2000], 
           attr(fit1.ext$fit, 'sim')$samples[[4]][['b_log10.sporeVolume:Life_styleObligateassociation']][1001:2000])
conf.95.d <- slp.d > quantile(slp.d, prob=0.025) & slp.d < quantile(slp.d, prob=0.975)
conf.50.d <- slp.d > quantile(slp.d, prob=0.25) & slp.d < quantile(slp.d, prob=0.75)
slp.ext <- bind_rows(data.frame(Life_style='free-living', response='extent', slope=slp.a, conf.50=conf.50.a, conf.95=conf.95.a), 
                   # data.frame(Life_style='opportunistic', response='extent', slope=slp.a+slp.b, conf.50=conf.50.b, conf.95=conf.95.b), 
                   data.frame(Life_style='facultative', response='extent', slope=slp.a+slp.c, conf.50=conf.50.c, conf.95=conf.95.c), 
                   data.frame(Life_style='obligate', response='extent', slope=slp.a+slp.d, conf.50=conf.50.d, conf.95=conf.95.d))
rm(slp.a, slp.b, slp.c, slp.d, conf.95.a, conf.50.a, conf.95.b, conf.50.b, conf.95.c, conf.50.c, conf.95.d, conf.50.d)

# area
slp.a <- c(attr(fit1.area$fit, 'sim')$samples[[1]][['b_log10.sporeVolume']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[2]][['b_log10.sporeVolume']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[3]][['b_log10.sporeVolume']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[4]][['b_log10.sporeVolume']][1001:2000])
conf.95.a <- slp.a > quantile(slp.a, prob=0.025) & slp.a < quantile(slp.a, prob=0.975)
conf.50.a <- slp.a > quantile(slp.a, prob=0.25) & slp.a < quantile(slp.a, prob=0.75)
# slp.b <- c(attr(fit1.area$fit, 'sim')$samples[[1]][['b_log10.sporeVolume:Life_styleB_Opportunisticassociation']][1001:2000], 
#            attr(fit1.area$fit, 'sim')$samples[[2]][['b_log10.sporeVolume:Life_styleB_Opportunisticassociation']][1001:2000], 
#            attr(fit1.area$fit, 'sim')$samples[[3]][['b_log10.sporeVolume:Life_styleB_Opportunisticassociation']][1001:2000], 
#            attr(fit1.area$fit, 'sim')$samples[[4]][['b_log10.sporeVolume:Life_styleB_Opportunisticassociation']][1001:2000])
# conf.95.b <- slp.b > quantile(slp.b, prob=0.025) & slp.b < quantile(slp.b, prob=0.975)
# conf.50.b <- slp.b > quantile(slp.b, prob=0.25) & slp.b < quantile(slp.b, prob=0.75)
slp.c <- c(attr(fit1.area$fit, 'sim')$samples[[1]][['b_log10.sporeVolume:Life_styleFacultativeassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[2]][['b_log10.sporeVolume:Life_styleFacultativeassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[3]][['b_log10.sporeVolume:Life_styleFacultativeassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[4]][['b_log10.sporeVolume:Life_styleFacultativeassociation']][1001:2000])
conf.95.c <- slp.c > quantile(slp.c, prob=0.025) & slp.c < quantile(slp.c, prob=0.975)
conf.50.c <- slp.c > quantile(slp.c, prob=0.25) & slp.c < quantile(slp.c, prob=0.75)
slp.d <- c(attr(fit1.area$fit, 'sim')$samples[[1]][['b_log10.sporeVolume:Life_styleObligateassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[2]][['b_log10.sporeVolume:Life_styleObligateassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[3]][['b_log10.sporeVolume:Life_styleObligateassociation']][1001:2000], 
           attr(fit1.area$fit, 'sim')$samples[[4]][['b_log10.sporeVolume:Life_styleObligateassociation']][1001:2000])
conf.95.d <- slp.d > quantile(slp.d, prob=0.025) & slp.d < quantile(slp.d, prob=0.975)
conf.50.d <- slp.d > quantile(slp.d, prob=0.25) & slp.d < quantile(slp.d, prob=0.75)
slp.area <- bind_rows(data.frame(Life_style='free-living', response='area', slope=slp.a, conf.50=conf.50.a, conf.95=conf.95.a), 
                   # data.frame(Life_style='opportunistic', response='area', slope=slp.a+slp.b, conf.50=conf.50.b, conf.95=conf.95.b), 
                   data.frame(Life_style='facultative', response='area', slope=slp.a+slp.c, conf.50=conf.50.c, conf.95=conf.95.c), 
                   data.frame(Life_style='obligate', response='area', slope=slp.a+slp.d, conf.50=conf.50.d, conf.95=conf.95.d))
rm(slp.a, slp.b, slp.c, slp.d, conf.95.a, conf.50.a, conf.95.b, conf.50.b, conf.95.c, conf.50.c, conf.95.d, conf.50.d)

# join
slp <- bind_rows(slp.ext, slp.area) %>% 
  mutate(Life_style = factor(Life_style, levels=c('free-living', 'opportunistic', 'facultative', 'obligate')))
slp %>% group_by(response, Life_style) %>% summarise(mean=mean(slope))


### plot
labs <- c(area = 'Range area', extent = 'Maximum distance')

# extent and area by volume
temp %>% 
  mutate(Life_style = recode(Life_style, `AFree living`='free-living', `B_Opportunistic association`='opportunistic', 
                             `Facultative association`='facultative', `Obligate association`='obligate'), 
         Life_style = factor(Life_style, levels=c('free-living', 'opportunistic', 'facultative', 'obligate'))) %>% 
  select(order, log10.sporeVolume, Life_style, geo_extent_prop, geo_area_prop) %>% 
  pivot_longer(cols=ends_with('prop'), names_to='response', values_to='value') %>% 
  ggplot(aes(x=log10.sporeVolume, y=car::logit(value), colour=Life_style)) + 
  geom_point(alpha=0.5, shape=16,size=4) + 
  stat_smooth(method='lm', alpha=0.25) + 
  labs(x=expression("Spore size ("*mu*"m³)"), y='Geographic extent (relative to maximum distance/area, logit)') + 
  facet_grid(rows=vars(response), scales='free_y') + 
  # scale_colour_manual(name='', values=colorRampPalette(c('black', 'orange'))(3)) +
  scale_color_manual(values= c("#FFFFB2","#FD8D3C", "#E31A1C"))+
  #scale_colour_brewer(palette = "YlOrRd") +
  theme(legend.position='none', 
        panel.background = element_blank(),
        axis.title = element_text(size=25),
        axis.text = element_text(size = 25),
         panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line(size = 0.25, linetype = 'longdash',
                                        colour = "gray"),
        strip.background = element_blank(),
        strip.text.y = element_blank()) -> p1

ggplot(slp, aes(x=slope,  fill=Life_style)) + 
  geom_density(alpha=0.9) + 
  geom_vline(xintercept=0) + 
  labs(x='Slope estimate', y='Density') + 
  facet_grid(rows=vars(response), labeller=labeller(response=labs)) + 
  # scale_colour_manual(name='', values=colorRampPalette(c('black', 'orange'))(3)) + 
  scale_fill_manual(values= c("#FFFFB2","#FD8D3C", "#E31A1C"))+
  #scale_colour_brewer(palette = "YlOrRd") + 
  scale_fill_brewer(palette = "YlOrRd")+
  theme(panel.background = element_blank(),
        legend.position = "none",
        strip.text = element_text(size=25),
        axis.text = element_text(size = 25),
        axis.title = element_text(size=25),
     panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line(size = 0.25, linetype = 'longdash',
                                        colour = "gray")
  )-> p2

p1 + p2



```


## Figure S1
```{r}
data.frame(table(AllFungi[!duplicated(AllFungi$names_to_use),c("phylum")]))
  
Ascos<-c(`Spore` = 17656,`No_spore` = 66169)
Basidios<-c(`Spore` = 7530,`No_spore` = 40875)
Chytrids<-c(`Spore` = 2,`No_spore` = 1216)
Glomeros<-c(`Spore` = 274,`No_spore` = 10)
Zygos<-c(`Spore` = 295,`No_spore` = 1071)

library(waffle)
library(RColorBrewer)

waffle(
  Ascos / 100, rows = 20, size = 0.5,
  colors = c("blue","light blue"),
  title = "Ascomycota",
  #xlab = "1 square = 100 Species",
  pad = 3,
  legend_pos = "none"
  #use_glyph = "star"
)->A

waffle(
  Basidios / 100, rows = 12, size = 0.5,
  colors = c("blue","light blue"),
  title = "Basidiomycota",
  #xlab = "1 square = 100 Species",
  pad = 3,
  legend_pos = "none"
  #use_glyph = "star"
)->B

waffle(
  Chytrids , rows = 20, size = 0.5,
  colors = c("blue","light blue"),
  title = "Chytridiomycota",
  #xlab = "1 square = 1 Species", 
  pad = 3,
  legend_pos = "none"
  #use_glyph = "star"
)->C

waffle(
  Glomeros , rows = 7, size = 0.5,
  colors = c("blue","light blue"),
  title = "Glomeromycota",
  #xlab = "1 square = 1 Species",
  pad = 3,
  legend_pos = "none"
  #use_glyph = "star"
)->G

waffle(
  Zygos , rows = 20, size = 0.5,
  colors = c("blue","light blue"),
  title = "Zygomycota",
  #xlab = "1 square = 1 Species", 
  pad = 3,
  legend_pos = "none"
  #use_glyph = "star"
)->Z

#Ploting them. First the "higher fungi (A and B) and then the lower fungi (C,G,Z)
iron(A, B)
iron(C,G,Z)
```


## Figure S2

Upper panel
```{r}

To_Analysis %>% 
  ungroup() %>% 
  filter(phylum=="Ascomycota" | phylum=="Basidiomycota") %>% 
  filter(SporeName=="Ascospores"|SporeName=="Basidiospores") %>% 
  mutate(Life_style = recode(Life_style, Plant = "Facultative association"), 
         Life_style = recode(Life_style, "B_Opportunistic association" = "AFree living")) %>% 
  ggplot()+
  aes(Life_style,SporeVolume,fill=Life_style)+
  geom_violin(alpha=0.8, draw_quantiles=c(0.25, 0.5, 0.75))+
  facet_grid(.~phylum, scales = "free_y")+
  scale_fill_brewer(palette="YlOrRd")+
  scale_y_log10(labels = scales::trans_format("log10", scales::math_format(10^.x)))+
  labs(y=expression("Spore size as volume ("*mu*"m³)"))+
    theme(panel.background = element_blank(),
    panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line(size = 0.25,linetype = 'longdash',
                                        colour = "gray"),
    axis.text.x = element_text(angle=45),
    legend.position = "none",
    axis.title.x=element_blank())
  
```

Lower panel
```{r}
To_Analysis %>% 
  ungroup() %>% 
  filter(!grepl("multi",SporeType,ignore.case = T)) %>% 
  filter(!SporeName=="Ascospores"|SporeName=="Basidiospores") %>% 
  mutate(Life_style = recode(Life_style, Plant = "Facultative association"), 
         Life_style = recode(Life_style, "B_Opportunistic association" = "AFree living")) %>% 
  ggplot()+
  aes(Life_style,SporeVolume,fill=Life_style)+
  geom_violin(alpha=0.8, draw_quantiles=c(0.25, 0.5, 0.75))+
  facet_grid(.~phylum, scales = "free_y")+
  scale_fill_brewer(palette="YlOrRd")+
  scale_y_log10(labels = scales::trans_format("log10", scales::math_format(10^.x)))+
  labs(y=expression("Spore size as volume ("*mu*"m³)"))+
    theme(panel.background = element_blank(),
    panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line(size = 0.25,linetype = 'longdash',
                                        colour = "gray"),
    axis.text.x = element_text(angle=45),
    legend.position = "none",
    axis.title.x=element_blank())
```

